---
site: workflowr::wflow_site
title: "Análisis Exploratorio"
author: "Marc Noguera-Julian, PhD. TreeTopUnder"
date: "`r format(Sys.time(), '%Y-%m-%d')`"
header-includes:
- \usepackage{fancyhdr}
- \usepackage{lipsum}
- \pagestyle{fancy}
- \fancyhead[CO,CE]{Microbioma intestinal canino en intervencion nutricional}
- \fancyfoot[LO,RE]{Marc Noguera-Julian, TreeTopUnder}
- \fancyhead[LE,RO]{\thepage}
- \fancyfoot[CO,CE]{}
- \usepackage[default]{sourcesanspro}
- \usepackage[T1]{fontenc}
- \usepackage{eso-pic,graphicx,transparent}
mainfont: Helvetica
output: html_document
---

```{r setup, include=TRUE, echo=F, message=FALSE, warning=FALSE, paged.print=FALSE, tidy=F}
knitr::opts_chunk$set(echo = FALSE)
knitr::opts_knit$set(root.dir = here::here())
```

## Calidad y *throughput*

```{r init variables, echo=F, message=FALSE, warning=FALSE, paged.print=FALSE, tidy=F}
suppressMessages(require(phyloseq))
suppressWarnings(require(dplyr))
ps_silva<-readRDS("data/ps_silva.rds")
min_throughput<-otu_table(ps_silva) %>% colSums() %>% min()
max_throughput<-otu_table(ps_silva) %>% colSums() %>% max()
```

Se han analizado un total de 53 muestras, de ellas se han eliminado 6 controles. En los gráficos se puede observar que el throughput medio oscila alrededor delos 38.000 reads por muestra (max: `r max_throughput`, min:`r min_throughput`), después de pasar filtros de calidad. Este es un throughput aceptable para los análisis planteados. Además, no se observan diferencias entre los grupos definidos por las variables respuesta consideradas ( *treatment, wet_dry, neutered* ) a nivel de throughput. Esta profundidad de secuenciación permite incluir todas las muestras en el análisis *downstream*.

*Deliverables: ThroughputByGroup_treatment_wet_dry_neutered.pdf, SequenceThroughput.xls*

```{r Control de Calidad, echo=F, message=FALSE, warning=FALSE, paged.print=FALSE, tidy=F}
ps_silva<-readRDS("data/ps_silva.rds")
suppressWarnings(require(ggpubr))
suppressWarnings(require(patchwork))
otu_counts<-colSums(otu_table(ps_silva))
sampleData<-data.frame(sample_data(ps_silva)) %>% 
  dplyr::mutate(counts=nonchim)


p0.treatment<-sampleData %>% 
  dplyr::select(treatment,counts) %>% 
  tidyr::drop_na() %>% 
  ggstatsplot::ggbetweenstats(x=treatment,y=counts,title="Throughput vs treatment",
                                            type = "non-parametric",ggtheme = ggplot2::theme_classic(),palette="Set1")
p1.dry_wet<-sampleData %>% 
  dplyr::select(dry_wet,counts) %>% 
  tidyr::drop_na() %>% 
  ggstatsplot::ggbetweenstats(x=dry_wet,y=counts,title="Throughput vs dry_wet",
                                            type = "robust",ggtheme = ggplot2::theme_classic(),palette="Dark2")

p2.neutered<-sampleData %>% 
  dplyr::select(neutered,counts) %>% 
  tidyr::drop_na() %>%
  ggstatsplot::ggbetweenstats(x=neutered,y=counts,title="Throughput vs neutered",
                                            type = "robust",ggtheme = ggplot2::theme_classic(),palette="Set2")

pT<-(p0.treatment / p1.dry_wet / p2.neutered )
ggsave(pT,filename="output/Deliverables/ThroughputByGroup_treatment_wet_dry_neutered.pdf",width=8,height=14)

p0.treatment

```

### Esfuerzo de secuenciación

*Deliverables: AlphaDiversityEstimation_minRR.xls, RarefactionCurve_treatment y RarefactionCurve_dry_wet_neutered.pdf*

Es importante observar si, con el esfuerzo de secuenciación realizado y el throughput final obtenido, la caracterización de la composición del microbioma es suficiente. Para eso, podemos representar gráficamente el numero de nuevos taxones conforme al número de secuencias obtenidas. Lógicamente, cuanto mas secuencias de calidad obtenidas de una muestra mas taxones diferentes podremos detectar. Aun así, si el esfuerzo de secuenciación es suficiente, deberla ser capaz de detectar la mayor parte de taxones diferentes en una muestra, de tal forma que aumentar aún mas el esfuerzo de secuenciación no deberia detectar nuevos taxones. Las curvas de rarefacción nos permiten saber a partir de que profundidad de secuenciación hemos saturado nuestra capacidad de detección o si tenemos *under-sampling* de nuestras muestras. En los datos analizados se observa un suficiente *sampling* de las muestras, llegando a *plateau* entre 10.000 y 20.000 secuencias, con lo cual, con los datos analizados se puede caracterizar debidamente la composición del microbioma.

```{r Curvas de rarefaccion, message=FALSE, warning=FALSE, warning=F, paged.print=FALSE,echo=F, results='hide',fig.keep='all',fig.cap="Curva de rarefacción representand el número de taxones (y-axis) vs el numero de secuencias obtenidas. Cada linea es una muestra."}
### W'll rarefy to even depth (40.000 counts)
suppressMessages(require(microbiome))
suppressMessages(require(vegan))
suppressMessages(require(picante))
suppressMessages(require(dplyr))
ps_silva <- readRDS("data/ps_silva.rds")
x.2.0<-ps_silva
# x.2.0<-filter_taxa(x.2.0, function (x) {sum(x > 0) > 1}, prune=TRUE)
x.2.0<-rarefy_even_depth(x.2.0,min(rowSums(t(otu_table(x.2.0)))),rngseed = 1234)
saveRDS(x.2.0,"data/ps_silva_RR.rds")
richnessDF<-alpha(x.2.0)
pd_df<-picante::pd(t(otu_table(x.2.0)),phy_tree(x.2.0),include.root = F)
richnessDF<-merge(richnessDF,pd_df,by="row.names")
richnessDF$SampleID<-richnessDF$Row.names
rownames(richnessDF)<-richnessDF$SampleID
richnessDF[1:5,c("observed","diversity_shannon","PD")] %>% knitr::kable()
xlsx::write.xlsx(richnessDF,file="output/Deliverables/AlphaDiversityEstimation_minRR.xls")
sampleData<-phyloseq::sample_data(ps_silva) %>% data.frame()

sampleData<-merge(sampleData,richnessDF,by.x="SampleID", by.y="SampleID") %>% dplyr::select(-Row.names.x)
rownames(sampleData)<-sampleData$SampleID
### Save Richness & Diversity parameters
ps_silva<-readRDS("data/ps_silva.rds")
if( ! grepl("diversity_shannon",colnames(sample_data(ps_silva)))){
  rownames(sampleData)<-sampleData$SampleID
  sample_data(ps_silva)<-sampleData
  saveRDS(ps_silva,"data/ps_silva.rds")
}

x.2.0<-ps_silva
pdf("output/Deliverables/RarefactionCurve_treatment.pdf")
myColors<-(RColorBrewer::brewer.pal(name = "Set2",n=3))
rarecurve(data.frame(t(otu_table(x.2.0))),step=200,col=myColors[as.factor(sampleData$treatment)],label=F,main="Observed Species vs Sequencing Depth vs treatment")
dev.off()
pdf("output/Deliverables/RarefactionCurve_dry_wet_neutered.pdf")
# x.2.0<-phyloseq::prune_samples( sample_data(x.2.0)$Especie == "Oso", x.2.0)
myColors<-(RColorBrewer::brewer.pal(name = "Dark2",n=3))
rarecurve(data.frame(t(otu_table(x.2.0))),step=200,col=myColors[as.factor(sampleData$dry_wet)],label=F,main="Observed Species vs Sequencing Depth vs dry_wet ")
# x.2.0<-phyloseq::prune_samples( sample_data(x.2.0)$Especie == "Oso", x.2.0)
myColors<-(RColorBrewer::brewer.pal(name = "Set1",n=3))
rarecurve(data.frame(t(otu_table(x.2.0))),step=200,col=myColors[as.factor(sampleData$neutered)],label=F,main="Observed Species vs Sequencing Depth vs neutered")
dev.off()
myColors<-(RColorBrewer::brewer.pal(name = "Set2",n=3))
rarecurve(data.frame(t(otu_table(x.2.0))),step=200,col=myColors[as.factor(sampleData$treatment)],label=F,main="Observed Species vs Sequencing Depth vs treatment")

# myColors<-(RColorBrewer::brewer.pal(name = "Accent",n=2))
# rarecurve(data.frame(t(otu_table(x.2.0))),step=200,col=myColors[as.factor(sampleData$sample_type)],label=F,main="Observed Species vs Sequencing Depth vs Sample type")
```

## Exportación de los resultados a formatos estándar

Con tal de poder reproducir o cambiar parte o todo el análisis en el futuro se comparten los ficheros de resultados intermedios que se han ido obteniendo en los diferentes pasos del análisis realizado.

### Abundancias totales y relativas

*Deliverables: GenusAbundance_counts.xls, GenusAbundance_RA.xls, PhylumAbundance_counts.xls, PhylumAbundance_RA.xls y ASVData.biom*

Se exportan los resultados como tabla excel. Esta tabla contiene los conteos para cada muestra de cada taxón detectado. Así mismo, para cada muestra se incluyen los identificadores de paciente y las variables respuesta consideradas ( *treatment, dry-wet, neutered*). Se crean 4 tablas, una a nivel de genero bacteriano y otra a nivel de *phylum* bacteriano, tanto a nivel de conteos absolutos como medidas de abundancia relativa(proporción). Se exportan también los datos al nivel taxonómico ASV en formato estándar \[BIOM.(<https://biom-format.org/>)\] (<https://biom-format.org/>)

```{r Obtencion y exportacion de resultados, echo=F, warning=F,results='hide',fig.keep='all'}
require(dplyr)
ps_silva<-readRDS("data/ps_silva.rds")
x.1.0 <- ps_silva 
x.1.0.genus<-phyloseq::tax_glom(x.1.0,taxrank="Genus",NArm = F)
x.1.0.phylum<-phyloseq::tax_glom(x.1.0,taxrank="Phylum",NArm = F)
x.1.0.genus.ra<-phyloseq::transform_sample_counts(x.1.0.genus, function(x) ((x/sum(x))))
x.1.0.phylum.ra<-phyloseq::transform_sample_counts(x.1.0.phylum, function(x) ((x/sum(x))))

phyloseq::psmelt(x.1.0.genus) %>%
  # select(SampleID,Phylum,Class,Order,Family,Genus,Especie,ID,,Abundance)%>%
  xlsx::write.xlsx(file="output/Deliverables/GenusAbundance_counts.xls")
phyloseq::psmelt(x.1.0.genus.ra) %>%
  # select(SampleID,Phylum,Class,Order,Family,Genus,Especie,Sexo,Edad,Edad_2020,Edad_2021,Region,Code,Year,Abundance)%>%
  dplyr::rename(`Relative Abundance`=Abundance)%>%
  xlsx::write.xlsx(file="output/Deliverables/GenusAbundance_RA.xls")

phyloseq::psmelt(x.1.0.phylum) %>%
  # select(SampleID,Phylum,Especie,Sexo,Edad,Edad_2020,Edad_2021,Region,Code,Year,Abundance)%>%
  xlsx::write.xlsx(file="output/Deliverables/PhylumAbundance_counts.xls")
phyloseq::psmelt(x.1.0.phylum.ra) %>%
  # select(SampleID,Phylum,Especie,Sexo,Edad,Edad_2020,Edad_2021,Region,Code,Year,Abundance)%>%
  dplyr::rename(`Relative Abundance`=Abundance)%>%
  xlsx::write.xlsx(file="output/Deliverables/PhylumAbundance_RA.xls")

biomformat::write_biom(biomformat::make_biom(phyloseq::otu_table(ps_silva),phyloseq::sample_data(ps_silva),phyloseq::tax_table(ps_silva)),biom_file="output/Deliverables/ps_silva.biom")
```

### Calculo y exportación de las relaciones filogenéticas

*Deliverables: Phylogenetic_tree.nwk y ASV_RepSequences.fas*

Para cada uno de los taxones (genero/Phylum) obtenidos de las secuencias crudas (Amplicon Sequence Variants o ASV) se obtiene una secuencia genética representativa dela region V3-V4 de 16S del gen rDNA para esa ASV. Estas secuencias se guardan en formato fasta y se utilizan para el análisis. El alineamiento del conjunto de secuencias permite calcular un arbol filogenético que se usará también para calcular la similitud entre las diferentes muestras, usando diferentes indices de beta-diversidad. Este arbol también se exporta en formato estándar Newick.

```{r relaciones filogeneticas, echo=F,message=FALSE, warning=FALSE, paged.print=FALSE,results='hide',fig.keep='all'}
load("data/raw/DADA2/DADA2_Rsession.RData")
ps_silva<-readRDS("data/ps_silva.rds")
ape::write.tree(phyloseq::phy_tree(ps_silva),file="output/Deliverables/Phylogenetic_tree.nwk")
DFForSeq<-colnames(seqtab.nochim)

file.remove("output/Deliverables/ASV_RepSequences.fas")
require(digest)
for(seq in DFForSeq){
  write(paste(">",digest(seq),"\n",seq,sep=""),file="output/Deliverables/ASV_RepSequences.fas",append=T)
}
```

## Clustering

### Abundancias relativas de géneros bacterianos por grupos.

Para representar gráficamente la composición del microbioma correspondiente a las muestras, se eliminan los géneros bacterianos que no están presente en al menos dos muestras y se representan solamente los 40 géneros mas abundantes en el dataset completo. de los taxones se representan las mismas en gráficos tipo *stacked barplots*.

*Deliverables: StackedBarplots_Genus.pdf y StackedBarplots_Genus_Mock.pdf DendrogramClustering.pdf*

```{r Representacion grafica de las abundancias de genero y filo bacteriano, echo=F,message=FALSE, results='hide',fig.keep='all',warning=FALSE, paged.print=FALSE, fig.cap="Abundancias relativas de los 30 generos bacterianos mayoritarios, segregado por la variables primarias y secundarias)"}
suppressWarnings(require(phyloseq))
suppressWarnings(require(phylosmith))
suppressWarnings(require(dplyr))
suppressWarnings(require(patchwork))
suppressWarnings(require(ggplot2))
suppressWarnings(require(ggthemes))
ps_silva<-readRDS("data/ps_silva.rds")
x.2.0<-ps_silva
x.2.0.genus<-tax_glom(x.2.0,taxrank="Genus",NArm = F)
wh0=genefilter_sample(x.2.0.genus,filterfun_sample(function(x) x>1), A=2) # Remove singletons
x.2.0.genus<-prune_taxa(wh0,x.2.0.genus)
x.2.0.genus = transform_sample_counts(x.2.0.genus, function(x) ((x/sum(x))))

ps.melt.x.2.0.genus<-x.2.0.genus %>%
  phyloseq::psmelt()

genus_for_order<-ps.melt.x.2.0.genus %>% 
  dplyr::select(Genus,Abundance) %>% 
  dplyr::group_by(Genus) %>% 
  summarise_at(vars(Abundance),list(mean_abundance=mean)) %>% 
  arrange(.,desc(mean_abundance)) %>% 
  dplyr::slice(1) %>% 
  select(Genus) 
myOrder<-reorder(ps.melt.x.2.0.genus[ps.melt.x.2.0.genus$Genus==genus_for_order$Genus,"SampleID"],ps.melt.x.2.0.genus[ps.melt.x.2.0.genus$Genus==genus_for_order$Genus,"Abundance"])

sampleData<-data.frame(phyloseq::sample_data(ps_silva))
sampleData$SampleID<-factor(sampleData$SampleID,levels=levels(myOrder),ordered = T)
rownames(sampleData)<-sampleData$SampleID
sample_data(x.2.0.genus)<-sampleData
### Keep the most abundant genus
meanByGenus<-aggregate(ps.melt.x.2.0.genus$Abundance,list(ps.melt.x.2.0.genus$Genus),mean)
colnames(meanByGenus)<-c("Genus","meanAbundance")
meanByGenus<-meanByGenus[with(meanByGenus, order(-meanAbundance, Genus)), ]
GenusToInclude<-meanByGenus[1:30,"Genus"]

p1<-phylosmith::phylogeny_profile(subset_taxa(x.2.0.genus,Genus%in% GenusToInclude),classification="Genus",relative_abundance=F,treatment="treatment",sample_labels = NULL,)+
 theme( panel.grid = element_blank(),axis.title= element_blank(), axis.text= element_blank())+
  theme(axis.title.x=element_blank(),axis.text.x=element_blank(),axis.ticks.x=element_blank())

p2<-phylosmith::phylogeny_profile(subset_taxa(x.2.0.genus,Genus%in% GenusToInclude),classification="Genus",relative_abundance=F,treatment="dry_wet",sample_labels = NULL,)+
 theme( panel.grid = element_blank(),axis.title= element_blank(), axis.text= element_blank())+
  theme(axis.title.x=element_blank(),axis.text.x=element_blank(),axis.ticks.x=element_blank())

p3<-phylosmith::phylogeny_profile(subset_taxa(x.2.0.genus,Genus%in% GenusToInclude),classification="Genus",relative_abundance=F,treatment="neutered",sample_labels = NULL,)+
 theme( panel.grid = element_blank(),axis.title= element_blank(), axis.text= element_blank())+
  theme(axis.title.x=element_blank(),axis.text.x=element_blank(),axis.ticks.x=element_blank())

p4<-phylosmith::phylogeny_profile(subset_taxa(x.2.0.genus,Genus%in% GenusToInclude),classification="Genus",relative_abundance=F,treatment="time_point",sample_labels = NULL,)+
 theme( panel.grid = element_blank(),axis.title= element_blank(), axis.text= element_blank())+
  theme(axis.title.x=element_blank(),axis.text.x=element_blank(),axis.ticks.x=element_blank())

p2
pT <- p1 / p2 | p3 / p4
ggsave(pT,filename="output/Deliverables/StackedBarplots_Genus.pdf",width=15,height=15)


```

### Dendrogramas & *Heatmap*

Estas abundancias relativas se usan también para calcular la similitud entre las diferentes muestras mediante distancias *weighted UniFrac* que se pueden utilizar para construir dendrogramas mediante *clústering*. En concreto se utiliza un algoritmo de *clustering* jerárquico bottom-up tipo Ward.D2. La diferencia entre la composición del microbioma de dos muestras es proporcional al camino vertical que habrá que recorrer en el árbol filogenético para ir de una muestra a la otra.

*Deliverables: WUnifrac_Distances.xls y ExploratoryHeatmap_allSamples.pdf*

Se observa un *clústering* de las muestras correspondientes a w2 y w1, enriquecido en muestras que corresponden al punto temporal t1. No se observan clusters de las muestras apareadas de un mismo animal, un resultado poco habitual. Tampoco la variable *neutered* tiente un efecto aparente en la estructura del microbioma.

```{R Dendrograms based on WUnifrac, message=FALSE, warning=FALSE, results='hide',fig.keep='all',echo=F,fig.cap="Dendrograma basado en distancia WUnifrac, considerando la composicion del microbioma fecal. "}
suppressPackageStartupMessages(require(ComplexHeatmap))
require(phyloseq)
suppressPackageStartupMessages(require(dplyr))
### Create Dendrograms based on WUnifrac Distance
x.2.0<-readRDS("data/ps_silva.rds")
x.2.0 = transform_sample_counts(x.2.0, function(x) ((x/sum(x))))
### Store WUnifrac distance from low-level ASV
myDistance<-phyloseq::distance(x.2.0,"wUniFrac")
myDist<-reshape2::melt(as.matrix(myDistance))
colnames(myDist)<-c("Sample1","Sample2","WUnifracDistance")
xlsx::write.xlsx(file="output/Deliverables/WUnifrac_Distances.xls",myDist)

# hclust(myDistance,method = "ward.D2")
### Prepare data for heatmap & dendrogram plot
x.2.0.genus<-phyloseq::tax_glom(x.2.0,taxrank="Genus",NArm = F)
ps.melt.x.2.0.genus<-x.2.0.genus %>%
  psmelt 
meanByGenus<-aggregate(ps.melt.x.2.0.genus$Abundance,list(ps.melt.x.2.0.genus$Genus),mean)
colnames(meanByGenus)<-c("Genus","meanAbundance")
meanByGenus<-meanByGenus[with(meanByGenus, order(-meanAbundance, Genus)), ]
GenusToInclude<-meanByGenus[1:90,"Genus"]


## Prepare Metadata
metadata_df<-sample_data(x.2.0.genus) %>% data.frame() %>% dplyr::select(treatment,dry_wet,neutered,dog_name,time_point)
# one_sample<-metadata_df %>% dplyr::select(dog_name) %>% dplyr::group_by(dog_name) %>% dplyr::summarise(n=n()) %>% dplyr::filter(n<=2) 
# metadata_df[metadata_df$Individuo %in% one_sample$Individuo,"Individuo"]<- "Otros"

## Prepare annotation
require(RColorBrewer)
### FActorize for plotting in controlled palette
metadata_df$treatment<-as.factor(metadata_df$treatment)
metadata_df$dry_wet<-as.factor(metadata_df$dry_wet)
metadata_df$neutered<-as.factor(metadata_df$neutered)
metadata_df$dog_name<-as.factor(metadata_df$dog_name)
metadata_df$time_point<-as.factor(metadata_df$time_point)
## Prepare color-scales for each variable
treatment_col<-colorRampPalette(brewer.pal(3,"Set2"))(length(levels(metadata_df$treatment)))
names(treatment_col)<-levels(metadata_df$treatment)
dry_wet_col<-colorRampPalette(brewer.pal(3,"Dark2"))(length(levels(metadata_df$dry_wet)))
names(dry_wet_col)<-levels(metadata_df$dry_wet)
neutered_col<-colorRampPalette(brewer.pal(3,"Set1"))(length(levels(metadata_df$neutered)))
names(neutered_col)<-levels(metadata_df$neutered)
dog_name_col<-colorRampPalette(brewer.pal(12,"Set3"))(length(levels(metadata_df$dog_name)))
names(dog_name_col)<-levels(metadata_df$dog_name)
time_point_col<-colorRampPalette(brewer.pal(12,"Set3"))(length(levels(metadata_df$time_point)))
names(time_point_col)<-levels(metadata_df$time_point)

ha <- ComplexHeatmap::HeatmapAnnotation(treatment=metadata_df$treatment,
                                        dry_wet=metadata_df$dry_wet,
                                        neutered=metadata_df$neutered,
                                        dog_name=metadata_df$dog_name,
                                        time_point=metadata_df$time_point,
                                        annotation_name_side = "left",
                                        annotation_legend_param = list(treatment = list(direction = "horizontal"),
                                                                       dry_wet=list(direction="horizontal"),
                                                                       neutered=list(direction="horizontal"),
                                                                       dog_name=list(direction="horizontal",ncol=3),
                                                                       time_point=list(direction="horizontal")),
                                        col=list(treatment=treatment_col,
                                                 dry_wet=dry_wet_col,
                                                 neutered=neutered_col,
                                                 dog_name=dog_name_col,
                                                 time_point=time_point_col))
## Prepare Heatmap Viz
suppressPackageStartupMessages(library(circlize))
col_fun = colorRamp2(c(0, 0.05, 1), c("steelblue", "white", "darkred"))

data_df<-(phyloseq::otu_table(subset_taxa(x.2.0.genus,Genus%in% GenusToInclude)))
taxa_df<-phyloseq::tax_table(subset_taxa(x.2.0.genus,Genus%in% GenusToInclude)) %>% data.frame() %>%  dplyr::select(Genus)
rownames(data_df)<-taxa_df$Genus

ht_list<-ComplexHeatmap::Heatmap(data_df,
                        cluster_columns = hclust(myDistance,method = "ward.D2"), cluster_rows = T,
                        column_title = "Muestra",row_title = "Taxa/Género",
                        show_row_names = T, show_column_names = F, show_row_dend = F,
                        row_names_gp = gpar(fontsize=6),
                        rect_gp = gpar(col = "white", lwd = 0.2),
                        top_annotation = ha, col=col_fun,
                        heatmap_legend_param = list(
                          title="relative abundance",
                          legend_height=unit(4,"cm"),
                          title_position="lefttop-rot",
                          direction="vertical",
                          legend_side="left"
                        ))
draw(ht_list,annotation_legend_side="bottom")
pdf("output/Deliverables/ExploratoryHeatmap_allSamples.pdf",width=9,height=9)
draw(ht_list,annotation_legend_side="bottom")
dev.off()
```

### Análisis global de beta-diversidad (Ordenación)

*Deliverables: NMDS_StressPlot.pdf, Adonis.xls y Ordination_AllSamples.pdf*

Para poder observar cambios estructurales en la composición del microbioma intestinal y su relación con las variables primarias y/o secundarias de interés, se utiliza el análisis de ordenación NMDS(*Non-metric dimensionality reduction*). El propósito de este análisis es el de hacer visualizable, en 2 dimensiones, la variabilidad existente en la composición del microbioma. De esta forma, cada punto del gráfico representa la composición del microbioma de una muestra, mientra su posición respecto a las otras muestras se relaciona con la similitud con esas muestras. La distancia entre dos muestras en el gráfico es la que mejor co-relaciona con la similitud real entre esas dos muestras teniendo en cuenta la composición del microbioma. Los valores de las variables primarias y/o secundarias se proyectan post-hoc sobre esas posiciones para revelar estructura en la composición que esté asociada a dichas variables. El análisis de ordenación se basa en el algoritmo NMDS y distancia Weighted-Unifrac(WUnifrac) y se evalúa la bondad/calidad de la ordenación mediante la comparación de la distancia/similitud real entre las dos muestras y al distancia proyectada en el plano bidimensional, recogiendo todas las comparaciones en el llamado *Stress Plot*.

Así mismo, se evalúa estadísticamente el peso de cada factor/variable analizada sobre la composición del microbioma mediante un test de ANOVA permutacional o PERMANOVA mediante la función vegan::adonis. Este test, proporciona un valor de significancia en relacion a si existe o no un efecto o asociación de la variable en cuestión sobre la composición ( *Pr(>F)* )un valor entre 0 y 1 relacionado con la magnitud de esta asociación en caso de ser significativa. ( *R2* ). Se realiza este test para cada variable y se seleccionan las variables que tienen un efecto significativo para construir el modelo multivariado.

En el análisis multivariado, se observa una influencia del factor *treatment* (R2=0.10, p\<0.05) y *time_point* (R2=0.076, p\<0.05) en la composición del microbioma, para el que según el test estadístico PERMANOVA estas dos variables explicarían conjuntamente un efecto del 17% de la composición de las muestras analizadas, apuntando a un efecto significativo de la intervención dietética en el microbioma.

```{r Analisis de Ordinacion, echo=FALSE, results='hide',fig.keep='all',fig.cap="Análisis de ordenacion NMDS/WUnifrac. En color se representa el punto temporal de la muestra.", message=FALSE, warning=FALSE, paged.print=FALSE,fig.keep='all'}
suppressWarnings(require(phyloseq))
suppressWarnings(require(phylosmith))
suppressWarnings(require(patchwork))
suppressWarnings(require(vegan))
x.2.0<-readRDS("data/ps_silva.rds")

set.seed(1234)
x.2.0.genus<-tax_glom(x.2.0,taxrank="Genus",NArm = F)
x.2.0.genus<-transform_sample_counts(x.2.0.genus, function(x) ((x/sum(x))))
sampleData<-data.frame(sample_data(x.2.0.genus))
rownames(sampleData)<-sampleData$SampleID
sample_data(x.2.0.genus)<-sampleData
(myOrd<-ordinate(x.2.0.genus,method="NMDS",distance="wunifrac"))

pdf("output/Deliverables/NMDS_StressPlot.pdf")
vegan::stressplot(myOrd)
dev.off()

#### PERMUTATIONAL ANOVA (adonis()) on WUnifrac distance
x.6.0.explanatory<-sampleData[,c("treatment","dry_wet","neutered","dog_name","time_point")]
x.6.0.response<-phyloseq::distance((x.2.0.genus),method="wunifrac")
myAdonis<-vegan::adonis2(x.6.0.response~treatment,data=x.6.0.explanatory)
file.remove("output/Deliverables/Adonis.xls")
xlsx::write.xlsx(file = "output/Deliverables/Adonis.xls",sheetName = "treatment",vegan::adonis2(x.6.0.response~treatment,data=x.6.0.explanatory))
### Perform ANOVA for non-NA in treatment
xlsx::write.xlsx(file = "output/Deliverables/Adonis.xls",sheetName = "dry_wet",vegan::adonis2(x.6.0.response~dry_wet,data=x.6.0.explanatory),append=T)
xlsx::write.xlsx(file = "output/Deliverables/Adonis.xls",sheetName = "neutered",vegan::adonis2(x.6.0.response~neutered,data=x.6.0.explanatory),append=T)
xlsx::write.xlsx(file = "output/Deliverables/Adonis.xls",sheetName = "dog_name",vegan::adonis2(x.6.0.response~dog_name,data=x.6.0.explanatory),append=T)
xlsx::write.xlsx(file = "output/Deliverables/Adonis.xls",sheetName = "time_point",vegan::adonis2(x.6.0.response~time_point,data=x.6.0.explanatory),append=T)

x.6.0.explanatory<-sampleData[,c("treatment","dry_wet","time_point")]
xlsx::write.xlsx(file = "output/Deliverables/Adonis.xls",sheetName = "multivariate",vegan::adonis2(x.6.0.response~.,data=x.6.0.explanatory),append=T)
### treatment and time_point have a significant effect on the fecal microbiome (~10%)
### Note that treatment and dry_wet variables are colinear/embedded. 
suppressMessages(require(mmtable2))
suppressMessages(require(tidyr))
vegan::adonis2(x.6.0.response~.,data=x.6.0.explanatory) %>% broom::tidy() %>% gridExtra::tableGrob() %>% grid::grid.draw()

####
if(! "diversity_shannon" %in% colnames(sample_data(x.2.0.genus))){
  richnessDF<-xlsx::read.xlsx(file="output/Deliverables/AlphaDiversityEstimation_noLAF.xls",sheetIndex = 1)
  rownames(richnessDF)<-richnessDF$NA.
  richnessDF$NA.<-NULL
  sampleData<-merge(sampleData,richnessDF,by.x="SampleID", by.y="row.names")
  rownames(sampleData)<-sampleData$SampleID
  sample_data(x.2.0.genus)<-sampleData
}

p.4.0.samples=plot_ordination(x.2.0.genus,myOrd)

### By treatment
require(ggplot2)
require(ggforce)
p0<-p.4.0.samples  + geom_point(aes(color=treatment)) +ggtitle("Unconstrained NMDS(WUnifrac) by treatment")+
  theme(panel.border=element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank(),axis.line=element_line(colour="black"),axis.text.x=element_text(size=16),axis.text.y=element_text(size=16))+
  theme(axis.title.x=element_text(size=14),axis.title.y=element_text(size=14))+
  theme(legend.text=element_text(size=12,face="italic"),legend.title=element_text(size=12))+
  theme(plot.title=element_text(lineheight=1,face="bold",size=15))+stat_ellipse(alpha=1,mapping=aes(color=treatment))+
  theme_classic() + theme(axis.line.x = element_line(colour = "black", 
                                    size = 1, linetype = "solid"), axis.line.y = element_line(colour = "black", 
                                    size = 1, linetype = "solid"), axis.text.x = element_text(size = 10), 
                                    axis.text.y = element_text(size = 10),
                                    axis.title.x = element_text(size = 10, 
                                        face = "bold"), axis.title.y = element_text(size = 10, 
                                        face = "bold"), legend.title = element_text(size = 10, 
                                        face = "bold"), legend.text = element_text(size = 8), 
                                        legend.spacing.x = unit(0.005, "npc"), 
                                        legend.key.size = unit(4,"mm"), legend.background = element_rect(fill = (alpha = 0))) + 
        labs(x = "NMDS Dimension 1", y = "NMDS Dimension 2") + 
        guides(fill = guide_legend(override.aes = list(size = 4)))

p1<-p.4.0.samples  + geom_point(aes(color=dry_wet)) +ggtitle("Unconstrained NMDS(WUnifrac) by dry_wet")+
  theme(panel.border=element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank(),axis.line=element_line(colour="black"),axis.text.x=element_text(size=16),axis.text.y=element_text(size=16))+
  theme(axis.title.x=element_text(size=14),axis.title.y=element_text(size=14))+
  theme(legend.text=element_text(size=12,face="italic"),legend.title=element_text(size=12))+
  theme(plot.title=element_text(lineheight=1,face="bold",size=15))+stat_ellipse(alpha=1,aes(color=dry_wet))+
  theme_classic() + theme(axis.line.x = element_line(colour = "black", 
                                    size = 1, linetype = "solid"), axis.line.y = element_line(colour = "black", 
                                    size = 1, linetype = "solid"), axis.text.x = element_text(size = 10), 
                                    axis.text.y = element_text(size = 10), axis.title.x = element_text(size = 10, 
                                    face = "bold"), axis.title.y = element_text(size = 10, 
                                    face = "bold"), legend.title = element_text(size = 10, 
                                    face = "bold"), legend.text = element_text(size = 8), 
                                     legend.spacing.x = unit(0.005, "npc"), legend.key.size = unit(4,"mm"), legend.background = element_rect(fill = (alpha = 0))) + 
        labs(x = "NMDS Dimension 1", y = "NMDS Dimension 2") + 
        guides(fill = guide_legend(override.aes = list(size = 4)))

p2<-p.4.0.samples  + geom_point(aes(color=neutered)) +ggtitle("Unconstrained NMDS(WUnifrac) by neutered")+
  theme(panel.border=element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank(),axis.line=element_line(colour="black"),axis.text.x=element_text(size=16),axis.text.y=element_text(size=16))+
  theme(axis.title.x=element_text(size=14),axis.title.y=element_text(size=14))+
  theme(legend.text=element_text(size=12,face="italic"),legend.title=element_text(size=12))+
  theme(plot.title=element_text(lineheight=1,face="bold",size=15))+stat_ellipse(alpha=1,aes(color=neutered))+
  theme_classic() + theme(axis.line.x = element_line(colour = "black", 
                                    size = 1, linetype = "solid"), axis.line.y = element_line(colour = "black", 
                                    size = 1, linetype = "solid"), axis.text.x = element_text(size = 10), 
                                    axis.text.y = element_text(size = 10), axis.title.x = element_text(size = 10, 
                                    face = "bold"), axis.title.y = element_text(size = 10, 
                                    face = "bold"), legend.title = element_text(size = 10, 
                                    face = "bold"), legend.text = element_text(size = 8), 
                                     legend.spacing.x = unit(0.005, "npc"), legend.key.size = unit(4,"mm"), legend.background = element_rect(fill = (alpha = 0))) + 
        labs(x = "NMDS Dimension 1", y = "NMDS Dimension 2") + 
        guides(fill = guide_legend(override.aes = list(size = 4)))

p3<-p.4.0.samples  + geom_point(aes(color=dog_name)) +ggtitle("Unconstrained NMDS(WUnifrac) by dog_name")+
  theme(panel.border=element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank(),axis.line=element_line(colour="black"),axis.text.x=element_text(size=16),axis.text.y=element_text(size=16))+
  theme(axis.title.x=element_text(size=14),axis.title.y=element_text(size=14))+
  theme(legend.text=element_text(size=12,face="italic"),legend.title=element_text(size=12))+
  theme(plot.title=element_text(lineheight=1,face="bold",size=15))+stat_ellipse(alpha=1,aes(color=dog_name))+
  theme_classic() + theme(axis.line.x = element_line(colour = "black", 
                                    size = 1, linetype = "solid"), axis.line.y = element_line(colour = "black", 
                                    size = 1, linetype = "solid"), axis.text.x = element_text(size = 10), 
                                    axis.text.y = element_text(size = 10), axis.title.x = element_text(size = 10, 
                                    face = "bold"), axis.title.y = element_text(size = 10, 
                                    face = "bold"), legend.title = element_text(size = 10, 
                                    face = "bold"), legend.text = element_text(size = 8), 
                                     legend.spacing.x = unit(0.005, "npc"), legend.key.size = unit(4,"mm"), legend.background = element_rect(fill = (alpha = 0))) + 
        labs(x = "NMDS Dimension 1", y = "NMDS Dimension 2") + 
        guides(fill = guide_legend(override.aes = list(size = 4)))

p4<-p.4.0.samples  + geom_point(aes(color=time_point)) +ggtitle("Unconstrained NMDS(WUnifrac) by time_point")+
  theme(panel.border=element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank(),axis.line=element_line(colour="black"),axis.text.x=element_text(size=16),axis.text.y=element_text(size=16))+
  theme(axis.title.x=element_text(size=14),axis.title.y=element_text(size=14))+
  theme(legend.text=element_text(size=12,face="italic"),legend.title=element_text(size=12))+
  theme(plot.title=element_text(lineheight=1,face="bold",size=15))+stat_ellipse(alpha=1,aes(color=time_point))+
  theme_classic() + theme(axis.line.x = element_line(colour = "black", 
                                    size = 1, linetype = "solid"), axis.line.y = element_line(colour = "black", 
                                    size = 1, linetype = "solid"), axis.text.x = element_text(size = 10), 
                                    axis.text.y = element_text(size = 10), axis.title.x = element_text(size = 10, 
                                    face = "bold"), axis.title.y = element_text(size = 10, 
                                    face = "bold"), legend.title = element_text(size = 10, 
                                    face = "bold"), legend.text = element_text(size = 8), 
                                     legend.spacing.x = unit(0.005, "npc"), legend.key.size = unit(4,"mm"), legend.background = element_rect(fill = (alpha = 0))) + 
        labs(x = "NMDS Dimension 1", y = "NMDS Dimension 2") + 
        guides(fill = guide_legend(override.aes = list(size = 4)))

 p0
pT<-p0 / p1 | p2 / p4


ggsave(pT,filename = "output/Deliverables/Ordination_AllSamples.pdf",width=15,height=15)

```

```{r ordination at BL, echo=FALSE, results='hide',fig.keep='all',fig.cap="Análisis de ordenacion NMDS/WUnifrac para muestras basales. En color se representa la especie del individuo.", message=FALSE, warning=FALSE, paged.print=FALSE,results='hide',fig.keep='all'}
suppressWarnings(require(phyloseq))
suppressWarnings(require(phylosmith))
suppressWarnings(require(patchwork))
suppressWarnings(require(vegan))
x.2.0<-readRDS("data/ps_silva.rds")

set.seed(1234)
x.2.0.genus<-tax_glom(x.2.0,taxrank="Genus",NArm = F) %>% 
  microbiome::transform(.,transform="compositional") %>% 
  phyloseq::subset_samples(.,time_point=="t0")

sampleData<-data.frame(sample_data(x.2.0.genus))
rownames(sampleData)<-sampleData$SampleID
sample_data(x.2.0.genus)<-sampleData
(myOrd<-ordinate(x.2.0.genus,method="NMDS",distance="wunifrac"))

pdf("output/Deliverables/NMDS_StressPlot_BL.pdf")
vegan::stressplot(myOrd)
dev.off()


#### PERMUTATIONAL ANOVA (adonis()) on WUnifrac distance (BL Samples)
x.6.0.explanatory<-sampleData[,c("treatment","dry_wet","neutered","dog_name")]
x.6.0.response<-phyloseq::distance((x.2.0.genus),method="wunifrac")
myAdonis<-vegan::adonis2(x.6.0.response~treatment,data=x.6.0.explanatory)
file.remove("output/Deliverables/Adonis_BL.xls")
xlsx::write.xlsx(file = "output/Deliverables/Adonis_BL.xls",sheetName = "treatment",vegan::adonis2(x.6.0.response~treatment,data=x.6.0.explanatory))
### Perform ANOVA for non-NA in treatment
xlsx::write.xlsx(file = "output/Deliverables/Adonis_BL.xls",sheetName = "dry_wet",vegan::adonis2(x.6.0.response~dry_wet,data=x.6.0.explanatory),append=T)
xlsx::write.xlsx(file = "output/Deliverables/Adonis_BL.xls",sheetName = "neutered",vegan::adonis2(x.6.0.response~neutered,data=x.6.0.explanatory),append=T)
xlsx::write.xlsx(file = "output/Deliverables/Adonis_BL.xls",sheetName = "dog_name",vegan::adonis2(x.6.0.response~dog_name,data=x.6.0.explanatory),append=T)
### These variables are not significant at BL.

p.4.0.samples=plot_ordination(x.2.0.genus,myOrd)

### By treatment
require(ggplot2)
require(ggforce)
p0<-p.4.0.samples  + geom_point(aes(color=treatment)) +ggtitle("Unconstrained NMDS(WUnifrac) by treatment")+
  theme(panel.border=element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank(),axis.line=element_line(colour="black"),axis.text.x=element_text(size=16),axis.text.y=element_text(size=16))+
  theme(axis.title.x=element_text(size=14),axis.title.y=element_text(size=14))+
  theme(legend.text=element_text(size=12,face="italic"),legend.title=element_text(size=12))+
  theme(plot.title=element_text(lineheight=1,face="bold",size=15))+stat_ellipse(alpha=1,mapping=aes(color=treatment))+
  theme_classic() + theme(axis.line.x = element_line(colour = "black", 
                                    size = 1, linetype = "solid"), axis.line.y = element_line(colour = "black", 
                                    size = 1, linetype = "solid"), axis.text.x = element_text(size = 10), 
                                    axis.text.y = element_text(size = 10),
                                    axis.title.x = element_text(size = 10, 
                                        face = "bold"), axis.title.y = element_text(size = 10, 
                                        face = "bold"), legend.title = element_text(size = 10, 
                                        face = "bold"), legend.text = element_text(size = 8), 
                                        legend.spacing.x = unit(0.005, "npc"), 
                                        legend.key.size = unit(4,"mm"), legend.background = element_rect(fill = (alpha = 0))) + 
        labs(x = "NMDS Dimension 1", y = "NMDS Dimension 2") + 
        guides(fill = guide_legend(override.aes = list(size = 4)))

p1<-p.4.0.samples  + geom_point(aes(color=dry_wet)) +ggtitle("Unconstrained NMDS(WUnifrac) by dry_wet")+
  theme(panel.border=element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank(),axis.line=element_line(colour="black"),axis.text.x=element_text(size=16),axis.text.y=element_text(size=16))+
  theme(axis.title.x=element_text(size=14),axis.title.y=element_text(size=14))+
  theme(legend.text=element_text(size=12,face="italic"),legend.title=element_text(size=12))+
  theme(plot.title=element_text(lineheight=1,face="bold",size=15))+stat_ellipse(alpha=1,aes(color=dry_wet))+
  theme_classic() + theme(axis.line.x = element_line(colour = "black", 
                                    size = 1, linetype = "solid"), axis.line.y = element_line(colour = "black", 
                                    size = 1, linetype = "solid"), axis.text.x = element_text(size = 10), 
                                    axis.text.y = element_text(size = 10), axis.title.x = element_text(size = 10, 
                                    face = "bold"), axis.title.y = element_text(size = 10, 
                                    face = "bold"), legend.title = element_text(size = 10, 
                                    face = "bold"), legend.text = element_text(size = 8), 
                                     legend.spacing.x = unit(0.005, "npc"), legend.key.size = unit(4,"mm"), legend.background = element_rect(fill = (alpha = 0))) + 
        labs(x = "NMDS Dimension 1", y = "NMDS Dimension 2") + 
        guides(fill = guide_legend(override.aes = list(size = 4)))

p2<-p.4.0.samples  + geom_point(aes(color=neutered)) +ggtitle("Unconstrained NMDS(WUnifrac) by neutered")+
  theme(panel.border=element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank(),axis.line=element_line(colour="black"),axis.text.x=element_text(size=16),axis.text.y=element_text(size=16))+
  theme(axis.title.x=element_text(size=14),axis.title.y=element_text(size=14))+
  theme(legend.text=element_text(size=12,face="italic"),legend.title=element_text(size=12))+
  theme(plot.title=element_text(lineheight=1,face="bold",size=15))+stat_ellipse(alpha=1,aes(color=neutered))+
  theme_classic() + theme(axis.line.x = element_line(colour = "black", 
                                    size = 1, linetype = "solid"), axis.line.y = element_line(colour = "black", 
                                    size = 1, linetype = "solid"), axis.text.x = element_text(size = 10), 
                                    axis.text.y = element_text(size = 10), axis.title.x = element_text(size = 10, 
                                    face = "bold"), axis.title.y = element_text(size = 10, 
                                    face = "bold"), legend.title = element_text(size = 10, 
                                    face = "bold"), legend.text = element_text(size = 8), 
                                     legend.spacing.x = unit(0.005, "npc"), legend.key.size = unit(4,"mm"), legend.background = element_rect(fill = (alpha = 0))) + 
        labs(x = "NMDS Dimension 1", y = "NMDS Dimension 2") + 
        guides(fill = guide_legend(override.aes = list(size = 4)))

p3<-p.4.0.samples  + geom_point(aes(color=dog_name)) +ggtitle("Unconstrained NMDS(WUnifrac) by dog_name")+
  theme(panel.border=element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank(),axis.line=element_line(colour="black"),axis.text.x=element_text(size=16),axis.text.y=element_text(size=16))+
  theme(axis.title.x=element_text(size=14),axis.title.y=element_text(size=14))+
  theme(legend.text=element_text(size=12,face="italic"),legend.title=element_text(size=12))+
  theme(plot.title=element_text(lineheight=1,face="bold",size=15))+stat_ellipse(alpha=1,aes(color=dog_name))+
  theme_classic() + theme(axis.line.x = element_line(colour = "black", 
                                    size = 1, linetype = "solid"), axis.line.y = element_line(colour = "black", 
                                    size = 1, linetype = "solid"), axis.text.x = element_text(size = 10), 
                                    axis.text.y = element_text(size = 10), axis.title.x = element_text(size = 10, 
                                    face = "bold"), axis.title.y = element_text(size = 10, 
                                    face = "bold"), legend.title = element_text(size = 10, 
                                    face = "bold"), legend.text = element_text(size = 8), 
                                     legend.spacing.x = unit(0.005, "npc"), legend.key.size = unit(4,"mm"), legend.background = element_rect(fill = (alpha = 0))) + 
        labs(x = "NMDS Dimension 1", y = "NMDS Dimension 2") + 
        guides(fill = guide_legend(override.aes = list(size = 4)))

p4<-p.4.0.samples  + geom_point(aes(color=time_point)) +ggtitle("Unconstrained NMDS(WUnifrac) by time_point")+
  theme(panel.border=element_blank(),panel.grid.major=element_blank(),panel.grid.minor=element_blank(),axis.line=element_line(colour="black"),axis.text.x=element_text(size=16),axis.text.y=element_text(size=16))+
  theme(axis.title.x=element_text(size=14),axis.title.y=element_text(size=14))+
  theme(legend.text=element_text(size=12,face="italic"),legend.title=element_text(size=12))+
  theme(plot.title=element_text(lineheight=1,face="bold",size=15))+stat_ellipse(alpha=1,aes(color=time_point))+
  theme_classic() + theme(axis.line.x = element_line(colour = "black", 
                                    size = 1, linetype = "solid"), axis.line.y = element_line(colour = "black", 
                                    size = 1, linetype = "solid"), axis.text.x = element_text(size = 10), 
                                    axis.text.y = element_text(size = 10), axis.title.x = element_text(size = 10, 
                                    face = "bold"), axis.title.y = element_text(size = 10, 
                                    face = "bold"), legend.title = element_text(size = 10, 
                                    face = "bold"), legend.text = element_text(size = 8), 
                                     legend.spacing.x = unit(0.005, "npc"), legend.key.size = unit(4,"mm"), legend.background = element_rect(fill = (alpha = 0))) + 
        labs(x = "NMDS Dimension 1", y = "NMDS Dimension 2") + 
        guides(fill = guide_legend(override.aes = list(size = 4)))


pT<-p0 / p1 | p2 / p4
ggsave(pT,filename = "output/Deliverables/Ordination_BLSamples.pdf",width=15,height=15)
```
