<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Marc Noguera-Julian, PhD." />

<meta name="date" content="2022-10-06" />

<title>Data Exploration</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">MedCars</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="201_exploratory_analysis.html">Data Exploration</a>
</li>
<li>
  <a href="210_baseline_analysis.html">Baseline Analysis</a>
</li>
<li>
  <a href="220_intervention_analysis.html">Intervention</a>
</li>
<li>
  <a href="230_MedDietScore.html">Mediterranean Diet</a>
</li>
<li>
  <a href="240_fatty_acids.html">Fatty Acids</a>
</li>
<li>
  <a href="250_PhysicalActivity.html">Physical Activity</a>
</li>
<li>
  <a href="260_FirmicutesBacteroides.html">Firmicutes/Bacteroidetes</a>
</li>
<li>
  <a href="270_PairingAnalysis.html">Paired Analysis</a>
</li>
<li>
  <a href="300_FunctionalAnalysis.html">Functional Analysis</a>
</li>
<li>
  <a href="320_BatchEffect.html">Batch Effects</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Data Exploration</h1>
<h4 class="author">Marc Noguera-Julian, PhD.</h4>
<h4 class="date">2022-10-06</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-10-06
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>2022_MedCars/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220720code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220720)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220720code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220720)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomMicrobialGenomicsIrsicaixaOrg2022MedCarstree136c267de0ff399e395309b721fd13d2fdc4cbe4targetblank136c267a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/MicrobialGenomics-IrsicaixaOrg/2022_MedCars/tree/136c267de0ff399e395309b721fd13d2fdc4cbe4" target="_blank">136c267</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomMicrobialGenomicsIrsicaixaOrg2022MedCarstree136c267de0ff399e395309b721fd13d2fdc4cbe4targetblank136c267a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/MicrobialGenomics-IrsicaixaOrg/2022_MedCars/tree/136c267de0ff399e395309b721fd13d2fdc4cbe4" target="_blank">136c267</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/cfu/
    Ignored:    data/internal/
    Ignored:    data/processed/
    Ignored:    data/ps_silva_RR.rds
    Ignored:    data/raw/
    Ignored:    output/.DS_Store
    Ignored:    output/Deliverables/
    Ignored:    output/mre_data/

Unstaged changes:
    Modified:   analysis/201_exploratory_analysis.Rmd
    Modified:   analysis/210_baseline_analysis.Rmd
    Modified:   analysis/215_Month6_analysis.Rmd
    Modified:   analysis/220_intervention_analysis.Rmd
    Modified:   analysis/230_MedDietScore.Rmd
    Modified:   analysis/260_FirmicutesBacteroides.Rmd
    Modified:   analysis/270_PairingAnalysis.Rmd
    Modified:   analysis/300_FunctionalAnalysis.Rmd
    Modified:   analysis/320_BatchEffect.Rmd
    Modified:   code/006_DataPreprocessing.Rmd
    Modified:   code/006_DataPreprocessing.html

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/201_exploratory_analysis.Rmd</code>) and HTML (<code>docs/201_exploratory_analysis.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics-IrsicaixaOrg/2022_MedCars/136c267de0ff399e395309b721fd13d2fdc4cbe4/docs/201_exploratory_analysis.html" target="_blank">136c267</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-10-06
</td>
<td>
final analysis
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics-IrsicaixaOrg/2022_MedCars/blob/4ccdb4cfd6d67a16afbcc4bf4c6e5f54a1b44bd3/analysis/201_exploratory_analysis.Rmd" target="_blank">4ccdb4c</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-09-30
</td>
<td>
additional analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics-IrsicaixaOrg/2022_MedCars/4ccdb4cfd6d67a16afbcc4bf4c6e5f54a1b44bd3/docs/201_exploratory_analysis.html" target="_blank">4ccdb4c</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-09-30
</td>
<td>
additional analysis
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics-IrsicaixaOrg/2022_MedCars/blob/9594b4b1eee834d1666dba133ec4ba2e68d2e3c9/analysis/201_exploratory_analysis.Rmd" target="_blank">9594b4b</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-09-23
</td>
<td>
maaslin and batch effects
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics-IrsicaixaOrg/2022_MedCars/9594b4b1eee834d1666dba133ec4ba2e68d2e3c9/docs/201_exploratory_analysis.html" target="_blank">9594b4b</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-09-23
</td>
<td>
maaslin and batch effects
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics-IrsicaixaOrg/2022_MedCars/blob/1c2dd47bf64409ce32bc85e3c96b5452d29e8fc1/analysis/201_exploratory_analysis.Rmd" target="_blank">1c2dd47</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-09-14
</td>
<td>
added analysis
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics-IrsicaixaOrg/2022_MedCars/940511c512797f8ef952c0b2e1224cd2cde0b938/docs/201_exploratory_analysis.html" target="_blank">940511c</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-08-19
</td>
<td>
Update 201_exploratory_analysis.html
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics-IrsicaixaOrg/2022_MedCars/601ea51e3b21ba0ec3870ef5008c6c222470b0fc/docs/201_exploratory_analysis.html" target="_blank">601ea51</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-08-18
</td>
<td>
several improvements
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/MicrobialGenomics-IrsicaixaOrg/2022_MedCars/6006450f6f52b347dfce2287c8f9feb88be7173f/docs/201_exploratory_analysis.html" target="_blank">6006450</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-08-02
</td>
<td>
Initial Commit
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="quality-and-throughput" class="section level2">
<h2>Quality and <em>throughput</em></h2>
<pre><code>suggested package selbal ✔suggested package DataExplorer ✔suggested package lme4 ✔suggested package merTools ✔</code></pre>
<p>A total of 988 samples have been sequenced. The plot shows that median throughput is around 1.13945^{4} (max: 2.8743^{4}, min:5664), after going through quality filters. This a limited but acceptable throughput. In addition there are no significant diffences between groups formed by primary variables (<em>Center</em>), although throughput from baseline(BL) samples is generally higher than that of Month 6 samples, although the magnitude of the differences is low.</p>
<p>There were a total of 494 of which 494(100%) had complete follow-up with baseline at month 6 fecal sample. These will be used for downstream analysis.</p>
<p><img src="figure/201_exploratory_analysis.Rmd/Quality%20Control-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Quality-Control-1">
Past versions of "Quality Control-1.png"
</button>
</p>
<div id="fig-no-spaces-Quality-Control-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics-IrsicaixaOrg/2022_MedCars/blob/136c267de0ff399e395309b721fd13d2fdc4cbe4/docs/figure/201_exploratory_analysis.Rmd/Quality Control-1.png" target="_blank">136c267</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-10-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics-IrsicaixaOrg/2022_MedCars/blob/6006450f6f52b347dfce2287c8f9feb88be7173f/docs/figure/201_exploratory_analysis.Rmd/Quality Control-1.png" target="_blank">6006450</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-08-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<p><img src="figure/201_exploratory_analysis.Rmd/Quality%20Control-2.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Quality-Control-2">
Past versions of "Quality Control-2.png"
</button>
</p>
<div id="fig-no-spaces-Quality-Control-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics-IrsicaixaOrg/2022_MedCars/blob/136c267de0ff399e395309b721fd13d2fdc4cbe4/docs/figure/201_exploratory_analysis.Rmd/Quality Control-2.png" target="_blank">136c267</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-10-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics-IrsicaixaOrg/2022_MedCars/blob/6006450f6f52b347dfce2287c8f9feb88be7173f/docs/figure/201_exploratory_analysis.Rmd/Quality Control-2.png" target="_blank">6006450</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-08-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="clustering" class="section level2">
<h2>Clustering</h2>
<div id="bacterial-genus-relative-abundance." class="section level3">
<h3>Bacterial Genus Relative abundance.</h3>
<p>In order to represent the bacterial composition of the analyzes samples in a simple way, we will use stacked barplots. Each vertical bar is a sample. Withi each bar, each color represents a different bacterial genus, according to the legend. The area of this color in a particular samples represents the relativa abundance of that genus in that particular sample.</p>
<p>Bacterial composition is clearly dominated by prevotella and bacteroides genus, as expected in healthy gut microbiomes. Other bacterial genera that are represented in the data are Faecalibacterium, Agathobacter and Bifidobacterium.</p>
<div class="figure" style="text-align: center">
<img src="figure/201_exploratory_analysis.Rmd/graphical%20representantion%20of%20bacterial%20genus-1.png" alt="Stacked barplots representing relative abundances for bacterial genus" width="672" />
<p class="caption">
Stacked barplots representing relative abundances for bacterial genus
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-graphical-representantion-of-bacterial-genus-1">
Past versions of "graphical representantion of bacterial genus-1.png"
</button>
</p>
<div id="fig-no-spaces-graphical-representantion-of-bacterial-genus-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics-IrsicaixaOrg/2022_MedCars/blob/136c267de0ff399e395309b721fd13d2fdc4cbe4/docs/figure/201_exploratory_analysis.Rmd/graphical representantion of bacterial genus-1.png" target="_blank">136c267</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-10-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics-IrsicaixaOrg/2022_MedCars/blob/9594b4b1eee834d1666dba133ec4ba2e68d2e3c9/docs/figure/201_exploratory_analysis.Rmd/graphical representantion of bacterial genus-1.png" target="_blank">9594b4b</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-09-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
</div>
<div id="beta-diversity" class="section level2">
<h2>Beta-Diversity</h2>
<div id="dendrograms-heatmap" class="section level3">
<h3>Dendrograms &amp; <em>Heatmap</em></h3>
<p>We use relative abundances to calculate similarity between samples using <em>weighted UniFrac</em> distances. These distances can then be used to perform hierarchical clustering using bottom-up Ward.D2 algorithm in order to group samples that are more similar. Similarity between microbiome composition is proportional to the length of the path need to walk through the dendrogram branches, so columns corresponding to more similar samples tend to be closer in the heatmap.</p>
<p>Again, it is observed that Prevotell and Bacteroides are the most predominant genus in these gut microbiome samples. Also, it can be seen that there is some clustering of the samples into 4 main groups, but that this clustering is not apparently related to any of the considered variables. The clustering is driven by the most abundant genus, as seen above: Bacteroides, prevotella, Faecalibacterium, Agathobacter, Bifidobacterium and several Ruminococcus genus.e</p>
<div class="figure" style="text-align: center">
<img src="figure/201_exploratory_analysis.Rmd/Dendrograms%20based%20on%20WUnifrac-1.png" alt="WUnifrac-based dendrogram(Ward.D2)" width="672" />
<p class="caption">
WUnifrac-based dendrogram(Ward.D2)
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Dendrograms-based-on-WUnifrac-1">
Past versions of "Dendrograms based on WUnifrac-1.png"
</button>
</p>
<div id="fig-no-spaces-Dendrograms-based-on-WUnifrac-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics-IrsicaixaOrg/2022_MedCars/blob/136c267de0ff399e395309b721fd13d2fdc4cbe4/docs/figure/201_exploratory_analysis.Rmd/Dendrograms based on WUnifrac-1.png" target="_blank">136c267</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-10-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics-IrsicaixaOrg/2022_MedCars/blob/9594b4b1eee834d1666dba133ec4ba2e68d2e3c9/docs/figure/201_exploratory_analysis.Rmd/Dendrograms based on WUnifrac-1.png" target="_blank">9594b4b</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-09-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics-IrsicaixaOrg/2022_MedCars/blob/6006450f6f52b347dfce2287c8f9feb88be7173f/docs/figure/201_exploratory_analysis.Rmd/Dendrograms based on WUnifrac-1.png" target="_blank">6006450</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-08-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="nmds-ordination" class="section level3">
<h3>NMDS Ordination</h3>
<p>In order to visualized global changes in the composition of gut microbiome and the relation of these changes with primary or secondary variable, we will use non-metric dimensionality reduction analysis (NMDS). The purpose of this analysis is to display microbiome composition structure in a two-dimension space. In this way, each point in the plot represent the microbial composition of a single sample and the distance between two points within this plot related to the dissimilarity of the microbial composition of those two samples.</p>
<p>NMDS Ordination space is dominated by a Bacteroides enrichment/Depletion axis both through NMDS1 and NMDS2. Asteroleplasma axis is dominating orthogonally to the first one. Other bacterial genus that have a dominating weight in the ordination space are Succinivibrio, Ruminococcus, Faecalibacterium and Erysipelotrichiae family. None of the considere variables show a correlation with ordination analysis, including timepoint, recruitment center, Alcoholice beverages (UBE), mediterranean diet score (MDScore) smoking or gender.</p>
<!-- Para poder observar cambios estructurales en la composición del microbioma intestinal y su relación con las variables primarias y/o secundarias de interés, se utiliza el análisis de ordenación NMDS(*Non-metric dimensionality reduction*). El propósito de este análisis es el de hacer visualizable, en 2 dimensiones, la variabilidad existente en la composición del microbioma. De esta forma, cada punto del gráfico representa la composición del microbioma de una muestra, mientra su posición respecto a las otras muestras se relaciona con la similitud con esas muestras. La distancia entre dos muestras en el gráfico es la que mejor co-relaciona con la similitud real entre esas dos muestras teniendo en cuenta la composición del microbioma. Los valores de las variables primarias y/o secundarias se proyectan post-hoc sobre esas posiciones para revelar estructura en la composición que esté asociada a dichas variables. El análisis de ordenación se basa en el algoritmo NMDS y distancia Weighted-Unifrac(WUnifrac) y se evalúa la bondad/calidad de la ordenación mediante la comparación de la distancia/similitud real entre las dos muestras y al distancia proyectada en el plano bidimensional, recogiendo todas las comparaciones en el llamado *Stress Plot*. -->
<p>Similarly we will statistically evaluate the weight of each of the variables on the composition of the gut microbiome using a permutational ANOVA or PERMANOVA. This tests provides a statistical assessment of this weight(<em>R2</em>) and its statistical significance (<em>Pr(&gt;F)</em>). Results indicate significant effect of factors such as Center, timepoint, Gender and Smoking but the magnitude of this effect is very low (&lt;1%). In addition, sample projection onto NMDS1 and NMDS does not reveal any significant difference. <!-- Así mismo, se evalúa estadísticamente el peso de cada factor/variable analizada sobre la composición del microbioma mediante un test de ANOVA permutacional o PERMANOVA mediante la función vegan::adonis. Este test, proporciona un valor de significancia en relacion a si existe o no un efecto o asociación de la variable en cuestión sobre la composición ( *Pr(>F)* )un valor entre 0 y 1 relacionado con la magnitud de esta asociación en caso de ser significativa. ( *R2* ). Se realiza este test para cada variable y se seleccionan las variables que tienen un efecto significativo para construir el modelo multivariado. --></p>
<!-- En el análisis multivariado, se observa una influencia del factor *treatment* (R2=0.10, p\<0.05) y *time_point* (R2=0.076, p\<0.05) en la composición del microbioma, para el que según el test estadístico PERMANOVA estas dos variables explicarían conjuntamente un efecto del 17% de la composición de las muestras analizadas, apuntando a un efecto significativo de la intervención dietética en el microbioma. -->
<div class="figure" style="text-align: center">
<img src="figure/201_exploratory_analysis.Rmd/Ordination%20Analysis-1.png" alt="NMDS/WUnifrac analysis. Color represents variable." width="672" />
<p class="caption">
NMDS/WUnifrac analysis. Color represents variable.
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Ordination-Analysis-1">
Past versions of "Ordination Analysis-1.png"
</button>
</p>
<div id="fig-no-spaces-Ordination-Analysis-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics-IrsicaixaOrg/2022_MedCars/blob/136c267de0ff399e395309b721fd13d2fdc4cbe4/docs/figure/201_exploratory_analysis.Rmd/Ordination Analysis-1.png" target="_blank">136c267</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-10-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics-IrsicaixaOrg/2022_MedCars/blob/9594b4b1eee834d1666dba133ec4ba2e68d2e3c9/docs/figure/201_exploratory_analysis.Rmd/Ordination Analysis-1.png" target="_blank">9594b4b</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-09-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics-IrsicaixaOrg/2022_MedCars/blob/6006450f6f52b347dfce2287c8f9feb88be7173f/docs/figure/201_exploratory_analysis.Rmd/Ordination Analysis-1.png" target="_blank">6006450</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-08-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="figure" style="text-align: center">
<img src="figure/201_exploratory_analysis.Rmd/Ordination%20Analysis-2.png" alt="NMDS/WUnifrac analysis. Color represents variable." width="672" />
<p class="caption">
NMDS/WUnifrac analysis. Color represents variable.
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-Ordination-Analysis-2">
Past versions of "Ordination Analysis-2.png"
</button>
</p>
<div id="fig-no-spaces-Ordination-Analysis-2" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics-IrsicaixaOrg/2022_MedCars/blob/136c267de0ff399e395309b721fd13d2fdc4cbe4/docs/figure/201_exploratory_analysis.Rmd/Ordination Analysis-2.png" target="_blank">136c267</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-10-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics-IrsicaixaOrg/2022_MedCars/blob/9594b4b1eee834d1666dba133ec4ba2e68d2e3c9/docs/figure/201_exploratory_analysis.Rmd/Ordination Analysis-2.png" target="_blank">9594b4b</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-09-23
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics-IrsicaixaOrg/2022_MedCars/blob/6006450f6f52b347dfce2287c8f9feb88be7173f/docs/figure/201_exploratory_analysis.Rmd/Ordination Analysis-2.png" target="_blank">6006450</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-08-02
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="pam-clustering" class="section level3">
<h3>PAM Clustering</h3>
<p>Using unsupervised clustering, we’ll try to see if there are clusters of microbial composition within the sample set. We’ll do this using Partition around Medoids (PAM) algorithm. According to the silhoutte coefficient values, there is no clustering effect in the gut microbiome composition. That is, the highest Silhouett coefficient (k=2) is &lt; 0.20, thus we can discard the hypothesis that there are two or more clusters formed by microbiome composition of the samples.</p>
<div class="figure" style="text-align: center">
<img src="figure/201_exploratory_analysis.Rmd/pam%20clustering-1.png" alt="Silhouette plot. X-axis represents the number of clusters being tested by the PAM algorithm" width="672" />
<p class="caption">
Silhouette plot. X-axis represents the number of clusters being tested by the PAM algorithm
</p>
</div>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-no-spaces-pam-clustering-1">
Past versions of "pam clustering-1.png"
</button>
</p>
<div id="fig-no-spaces-pam-clustering-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics-IrsicaixaOrg/2022_MedCars/blob/136c267de0ff399e395309b721fd13d2fdc4cbe4/docs/figure/201_exploratory_analysis.Rmd/pam clustering-1.png" target="_blank">136c267</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-10-06
</td>
</tr>
<tr>
<td>
<a href="https://github.com/MicrobialGenomics-IrsicaixaOrg/2022_MedCars/blob/601ea51e3b21ba0ec3870ef5008c6c222470b0fc/docs/figure/201_exploratory_analysis.Rmd/pam clustering-1.png" target="_blank">601ea51</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-08-18
</td>
</tr>
</tbody>
</table>
</div>
</div>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre><code>R version 4.1.1 (2021-08-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] metar_0.1.5           tidyr_1.2.0           mmtable2_0.1.3       
 [4] vegan_2.6-2           lattice_0.20-45       permute_0.9-7        
 [7] ComplexHeatmap_2.10.0 ggthemes_4.2.4        phylosmith_1.0.6     
[10] patchwork_1.1.1       ggpubr_0.4.0          ggplot2_3.3.6        
[13] testthat_3.1.4        dplyr_1.0.9           phyloseq_1.38.0      

loaded via a namespace (and not attached):
  [1] coda_0.19-4            ragg_1.2.2             bit64_4.0.5           
  [4] knitr_1.39             wesanderson_0.3.6      data.table_1.14.2     
  [7] DataExplorer_0.8.2     RCurl_1.98-1.8         doParallel_1.0.17     
 [10] generics_0.1.3         BiocGenerics_0.40.0    callr_3.7.1           
 [13] usethis_2.1.6          proxy_0.4-27           future_1.27.0         
 [16] correlation_0.8.1      bit_4.0.4              tzdb_0.3.0            
 [19] xml2_1.3.3             lubridate_1.8.0        httpuv_1.6.5          
 [22] assertthat_0.2.1       viridis_0.6.2          xfun_0.31             
 [25] rJava_1.0-6            jquerylib_0.1.4        evaluate_0.15         
 [28] promises_1.2.0.1       fansi_1.0.3            caTools_1.18.2        
 [31] igraph_1.3.5           DBI_1.1.3              htmlwidgets_1.5.4     
 [34] stats4_4.1.1           paletteer_1.4.0        purrr_0.3.4           
 [37] ellipsis_0.3.2         backports_1.4.1        insight_0.18.0        
 [40] prismatic_1.1.0        RcppParallel_5.1.5     vctrs_0.4.1           
 [43] Biobase_2.54.0         remotes_2.4.2          here_1.0.1            
 [46] abind_1.4-5            cachem_1.0.6           withr_2.5.0           
 [49] ggforce_0.3.3          vroom_1.5.7            prettyunits_1.1.1     
 [52] cluster_2.1.3          ape_5.6-2              crayon_1.5.1          
 [55] pkgconfig_2.0.3        labeling_0.4.2         units_0.8-0           
 [58] tweenr_1.0.2           GenomeInfoDb_1.30.1    nlme_3.1-158          
 [61] statsExpressions_1.3.2 pkgload_1.3.0          blme_1.0-5            
 [64] devtools_2.4.4         rlang_1.0.6            globals_0.15.1        
 [67] lifecycle_1.0.2        miniUI_0.1.1.1         MatrixModels_0.5-0    
 [70] selbal_0.1.0           rprojroot_2.0.3        polyclip_1.10-0       
 [73] matrixStats_0.62.0     datawizard_0.4.1       Matrix_1.4-1          
 [76] carData_3.0-5          Rhdf5lib_1.16.0        boot_1.3-28           
 [79] GlobalOptions_0.1.2    whisker_0.4            processx_3.7.0        
 [82] rjson_0.2.21           png_0.1-7              viridisLite_0.4.1     
 [85] parameters_0.18.1      bitops_1.0-7           KernSmooth_2.23-20    
 [88] rhdf5filters_1.6.0     Biostrings_2.62.0      workflowr_1.7.0       
 [91] shape_1.4.6            classInt_0.4-7         stringr_1.4.1         
 [94] arm_1.12-2             parallelly_1.32.1      rstatix_0.7.0         
 [97] S4Vectors_0.32.4       ggsignif_0.6.3         scales_1.2.1          
[100] memoise_2.0.1          magrittr_2.0.3         plyr_1.8.7            
[103] gplots_3.1.3           zlibbioc_1.40.0        compiler_4.1.1        
[106] RColorBrewer_1.1-3     clue_0.3-61            lme4_1.1-30           
[109] snakecase_0.11.0       cli_3.4.1              ade4_1.7-19           
[112] XVector_0.34.0         urlchecker_1.0.1       listenv_0.8.0         
[115] pbapply_1.5-0          ps_1.7.1               ggside_0.2.1          
[118] MASS_7.3-58            mgcv_1.8-40            tidyselect_1.1.2      
[121] stringi_1.7.8          forcats_0.5.1          textshaping_0.3.6     
[124] highr_0.9              yaml_2.3.5             ggrepel_0.9.1         
[127] sass_0.4.2             tools_4.1.1            parallel_4.1.1        
[130] circlize_0.4.15        rstudioapi_0.13        foreach_1.5.2         
[133] logging_0.10-108       git2r_0.30.1           janitor_2.1.0         
[136] gridExtra_2.3          farver_2.1.1           Rtsne_0.16            
[139] ggraph_2.0.5           digest_0.6.29          shiny_1.7.2           
[142] networkD3_0.4          Rcpp_1.0.9             car_3.1-0             
[145] broom_1.0.0            xlsx_0.6.5             BayesFactor_0.9.12-4.4
[148] performance_0.9.1      later_1.3.0            sf_1.0-7              
[151] effectsize_0.7.0       colorspace_2.0-3       brio_1.1.3            
[154] fs_1.5.2               IRanges_2.28.0         splines_4.1.1         
[157] rematch2_2.1.2         graphlayouts_0.8.0     xlsxjars_0.6.1        
[160] multtest_2.50.0        sessioninfo_1.2.2      systemfonts_1.0.4     
[163] xtable_1.8-4           jsonlite_1.8.0         nloptr_2.0.3          
[166] tidygraph_1.2.1        zeallot_0.1.0          gt_0.6.0              
[169] R6_2.5.1               profvis_0.3.7          broom.mixed_0.2.9.4   
[172] pillar_1.8.1           htmltools_0.5.3        mime_0.12             
[175] glue_1.6.2             fastmap_1.1.0          minqa_1.2.4           
[178] class_7.3-20           codetools_0.2-18       pkgbuild_1.3.1        
[181] mvtnorm_1.1-3          furrr_0.3.0            utf8_1.2.2            
[184] bslib_0.4.0            tibble_3.1.8           logger_0.2.2          
[187] gtools_3.9.3           magick_2.7.3           survival_3.3-1        
[190] rmarkdown_2.14         desc_1.4.1             biomformat_1.22.0     
[193] merTools_0.5.2         munsell_0.5.0          GetoptLong_1.0.5      
[196] e1071_1.7-11           rhdf5_2.38.1           GenomeInfoDbData_1.2.7
[199] iterators_1.0.14       ggstatsplot_0.9.3      reshape2_1.4.4        
[202] gtable_0.3.1           bayestestR_0.12.1     </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
