<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Marc Noguera-Julian, PhD." />

<meta name="date" content="2022-08-02" />

<title>Microbiome association with Mediterranean diet</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">MedCars</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="201_exploratory_analysis.html">Data Exploration</a>
</li>
<li>
  <a href="210_baseline_analysis.html">Baseline Analysis</a>
</li>
<li>
  <a href="220_intervention_analysis.html">Intervention</a>
</li>
<li>
  <a href="230_MedDietScore.html">Mediterranean Diet</a>
</li>
<li>
  <a href="901_Abstract.html">Abstract</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Microbiome association with Mediterranean diet</h1>
<h4 class="author">Marc Noguera-Julian, PhD.</h4>
<h4 class="date">2022-08-02</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-08-02
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>2022_MedCars/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown is ignored by Git. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220720code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220720)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220720code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220720)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrong989ad9b"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> 989ad9b </a>
</p>
</div>
<div id="strongRepositoryversionstrong989ad9b" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version 989ad9b. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/201_exploratory_analysis.Rmd
    Ignored:    analysis/210_baseline_analysis.Rmd
    Ignored:    analysis/220_intervention_analysis.Rmd
    Ignored:    analysis/230_MedDietScore.Rmd
    Ignored:    analysis/901_Abstract.Rmd
    Ignored:    data/.DS_Store
    Ignored:    data/cfu/
    Ignored:    data/internal/
    Ignored:    data/processed/
    Ignored:    data/ps_silva_RR.rds
    Ignored:    data/raw/
    Ignored:    output/Deliverables/
    Ignored:    output/mre_data/

Untracked files:
    Untracked:  code/006_DataPreprocessing.Rmd
    Untracked:  code/006_DataPreprocessing.html
    Untracked:  code/008_PopulateMRE.Rmd
    Untracked:  code/02_Exploratory_v1.Rmd
    Untracked:  code/03_Longitudinal_v1.Rmd
    Untracked:  code/04_Conclusiones_v1.Rmd

Unstaged changes:
    Modified:   .gitignore
    Modified:   analysis/_site.yml
    Modified:   analysis/index.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">
<p>
There are no past versions. Publish this analysis with <code>wflow_publish()</code> to start tracking its development.
</p>
<hr>
</div>
</div>
</div>
<div id="meddiet-score---baseline" class="section level2">
<h2>MedDiet Score - Baseline</h2>
<p>We will use Mediterranean Diet Score (MedDietScore) to assess whether participant’s adherence to mediterranean diet (MD) shows any association with microbiome composition or derived parameters. MedDietScore is an integer score ranging from 0 to 14, obtained from a simplified dietary questionnaire (PREDIMED). A medDietScore of 14 indicates maximal adherence to MD while 0 indicates no adherence at all.</p>
<div id="alpha-diversity---baseline" class="section level3">
<h3>Alpha Diversity - Baseline</h3>
<p>We will check, at baseline, if MedDietScore shows any association with Alpha diversity measures (Chao1, Shannon, PD). Results indicate no correlation between any of the alpha-diversity parameters and MedDietScore.</p>
<pre><code>suggested package selbal ✔suggested package DataExplorer ✔suggested package lme4 ✔suggested package merTools ✔</code></pre>
<p><img src="figure/230_MedDietScore.Rmd/BL%20Alpha%20Diversity%20Chunk%201-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/230_MedDietScore.Rmd/BL%20Alpha%20Diversity%20Chunk%201-2.png" width="672" style="display: block; margin: auto;" /><img src="figure/230_MedDietScore.Rmd/BL%20Alpha%20Diversity%20Chunk%201-3.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="nmds-ordination---baseline" class="section level3">
<h3>NMDS Ordination - Baseline</h3>
<p>We want to check, for baseline samples, if there is any relationship between gut microbiome composition and MedDietScore. For that we perform an NMDS ordination on baseline samples (see baseline analysis) and use each sample NMDS1 and NMDS2 axis coordinate to check correlation with MedDietScore.</p>
<pre><code>suggested package selbal ✔suggested package DataExplorer ✔suggested package lme4 ✔suggested package merTools ✔</code></pre>
<p><img src="figure/230_MedDietScore.Rmd/BL%20Ordination%20chunk%201-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="mdscore-genus-correlation" class="section level3">
<h3>MDScore-Genus correlation</h3>
<p>Apparently, there is a significant correlation between NMDS1 and MedDiet Score. Note, that according to baseline exploration, NMDS1 loads were clearly dominated by Bacteroides presence. In addition we will examine also other genus within this dataset. We will select those that are present in at least 20% of the samples. When we examine association between this genus and other and MedDiet Score we detect:</p>
<ul>
<li>Weak but significant correlations between MDScore and Lachnospira and Ruminococcus Genera</li>
<li>Weak but significant inverse correlations between MDScore and Bifidobacterium and Oscillibacter genera.</li>
</ul>
<p>Interestingly, Lachnospira genus showed to be enriched after nutrititional intervention in the Intervention Group(C01) but not in the control group(C02). Lachnospiraceae family species are butyrate producers, which specifically habit mammal digestive tract. Their role in human health is [controverted] (<a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7232163/" class="uri">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7232163/</a>)</p>
<p>Lachnospira was also found to be <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7401267/">enriched in higher MD-adherent healthy dults</a>. Lachnospira spp also increased in lean individuals upon <a href="https://journals.lww.com/acsm-msse/Fulltext/2018/04000/Exercise_Alters_Gut_Microbiota_Composition_and.14.aspx">physical exercising program</a>, linked to an increase in the detection of butyrate and other SCFAs in feces. Other studies have linked Lachnospira to physical activity or active lifestyle.</p>
<p>Lachnospira has also been linked to the consumption of high-fiber diet in <a href="https://doi.org/10.1017/S0007114518002465">two independent populations</a> and a randomized clinical study in healthy adults shifting from <a href="https://pubmed.ncbi.nlm.nih.gov/28179226/">whole grains to refined ones</a></p>
<pre><code>suggested package selbal ✔suggested package DataExplorer ✔suggested package lme4 ✔suggested package merTools ✔</code></pre>
<div class="figure" style="text-align: center">
<img src="figure/230_MedDietScore.Rmd/BL%20Ordination%20chunk%202-1.png" alt="Correlation heatmap showin correlation values between selected bacterial genus and MDScore values. Blank squares indicate marginally significant correlations (0.05&lt;p&lt;0.1) while values indicate correlation values (r) for significant correlations(p&lt;0.05)" width="672" />
<p class="caption">
Correlation heatmap showin correlation values between selected bacterial genus and MDScore values. Blank squares indicate marginally significant correlations (0.05&lt;p&lt;0.1) while values indicate correlation values (r) for significant correlations(p&lt;0.05)
</p>
</div>
</div>
</div>
<div id="intervention" class="section level2">
<h2>Intervention</h2>
<div id="meddietscore-changes" class="section level3">
<h3>MedDietScore Changes</h3>
<p>We want to assess it MedDietScore changes after the intervention and if there are any differences between groups in order to assess the effectivity of the intervention. We can see that the median change in the Control Group is 0 while in the intervention group is +1, thus showing an improvement in MD adherence in the intervention group and significant differences in these changes between both groups.</p>
<pre><code>suggested package selbal ✔suggested package DataExplorer ✔suggested package lme4 ✔suggested package merTools ✔</code></pre>
<p><img src="figure/230_MedDietScore.Rmd/Intervention%20chunk%201-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="md-changes-alpha" class="section level3">
<h3>MD Changes &amp; Alpha</h3>
<p>We want to see if there is any relationship between the magnitude of change in MD adherence (reported through MDScore) and the changes in alpha-diversity. For that we will calculate the relative changes in alpha diversity (Shannon only) for each of the participants, defined as (ShannonMonth6 - ShannonBL)/(ShannonMonth6 + ShannonBL) and check statistical association with changes in MDScore. Results indicated that there is no significant relationship.</p>
<pre><code>suggested package selbal ✔suggested package DataExplorer ✔suggested package lme4 ✔suggested package merTools ✔</code></pre>
<p><img src="figure/230_MedDietScore.Rmd/Intervention%20chunk%202-1.png" width="672" style="display: block; margin: auto;" /> ### MD Changes &amp; Beta</p>
<p>We want to see if there is any relationship between the magnitude of change in MD adherence (reported through MDScore) and the magnitude of changes in the composition of the gut microbiome of the participants. For that we will calculate the relative changes using Weighted-Unifrac distance for each of the participants, and check statistical association with changes in MDScore. Results indicated that there is no significant relationship (p=0.08)</p>
<pre><code>suggested package selbal ✔suggested package DataExplorer ✔suggested package lme4 ✔suggested package merTools ✔</code></pre>
<p><img src="figure/230_MedDietScore.Rmd/Intervention%20chunk%203-1.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre><code>R version 4.1.1 (2021-08-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] metar_0.1.5       testthat_3.1.4    ggstatsplot_0.9.3 vegan_2.6-2      
[5] lattice_0.20-45   permute_0.9-7     patchwork_1.1.1   phylosmith_1.0.6 
[9] phyloseq_1.38.0  

loaded via a namespace (and not attached):
  [1] coda_0.19-4            tidyr_1.2.0            ggplot2_3.3.6         
  [4] bit64_4.0.5            knitr_1.39             rpart_4.1.16          
  [7] wesanderson_0.3.6      data.table_1.14.2      DataExplorer_0.8.2    
 [10] RCurl_1.98-1.7         doParallel_1.0.17      generics_0.1.3        
 [13] BiocGenerics_0.40.0    callr_3.7.1            usethis_2.1.6         
 [16] proxy_0.4-27           future_1.27.0          correlation_0.8.1     
 [19] bit_4.0.4              tzdb_0.3.0             lubridate_1.8.0       
 [22] httpuv_1.6.5           assertthat_0.2.1       viridis_0.6.2         
 [25] xfun_0.31              jquerylib_0.1.4        evaluate_0.15         
 [28] promises_1.2.0.1       fansi_1.0.3            igraph_1.3.4          
 [31] DBI_1.1.3              htmlwidgets_1.5.4      stats4_4.1.1          
 [34] paletteer_1.4.0        purrr_0.3.4            ellipsis_0.3.2        
 [37] corrplot_0.92          dplyr_1.0.9            ggpubr_0.4.0          
 [40] backports_1.4.1        insight_0.18.0         prismatic_1.1.0       
 [43] deldir_1.0-6           RcppParallel_5.1.5     vctrs_0.4.1           
 [46] Biobase_2.54.0         remotes_2.4.2          here_1.0.1            
 [49] abind_1.4-5            cachem_1.0.6           withr_2.5.0           
 [52] ggforce_0.3.3          checkmate_2.1.0        vroom_1.5.7           
 [55] prettyunits_1.1.1      cluster_2.1.3          ape_5.6-2             
 [58] crayon_1.5.1           pkgconfig_2.0.3        labeling_0.4.2        
 [61] units_0.8-0            tweenr_1.0.2           GenomeInfoDb_1.30.1   
 [64] nlme_3.1-158           statsExpressions_1.3.2 pkgload_1.3.0         
 [67] nnet_7.3-17            blme_1.0-5             devtools_2.4.4        
 [70] rlang_1.0.4            globals_0.15.1         lifecycle_1.0.1       
 [73] miniUI_0.1.1.1         selbal_0.1.0           rprojroot_2.0.3       
 [76] polyclip_1.10-0        matrixStats_0.62.0     datawizard_0.4.1      
 [79] Matrix_1.4-1           carData_3.0-5          Rhdf5lib_1.16.0       
 [82] boot_1.3-28            base64enc_0.1-3        GlobalOptions_0.1.2   
 [85] processx_3.7.0         png_0.1-7              viridisLite_0.4.0     
 [88] rjson_0.2.21           parameters_0.18.1      bitops_1.0-7          
 [91] KernSmooth_2.23-20     rhdf5filters_1.6.0     Biostrings_2.62.0     
 [94] workflowr_1.7.0        shape_1.4.6            classInt_0.4-7        
 [97] stringr_1.4.0          arm_1.12-2             parallelly_1.32.1     
[100] jpeg_0.1-9             rstatix_0.7.0          S4Vectors_0.32.4      
[103] ggsignif_0.6.3         scales_1.2.0           memoise_2.0.1         
[106] magrittr_2.0.3         plyr_1.8.7             zlibbioc_1.40.0       
[109] compiler_4.1.1         RColorBrewer_1.1-3     clue_0.3-61           
[112] lme4_1.1-30            snakecase_0.11.0       cli_3.3.0             
[115] ade4_1.7-19            XVector_0.34.0         urlchecker_1.0.1      
[118] listenv_0.8.0          ps_1.7.1               htmlTable_2.4.1       
[121] Formula_1.2-4          MASS_7.3-58            mgcv_1.8-40           
[124] ggside_0.2.1           tidyselect_1.1.2       stringi_1.7.8         
[127] forcats_0.5.1          highr_0.9              yaml_2.3.5            
[130] latticeExtra_0.6-30    ggrepel_0.9.1          grid_4.1.1            
[133] sass_0.4.2             tools_4.1.1            parallel_4.1.1        
[136] circlize_0.4.15        rstudioapi_0.13        foreign_0.8-82        
[139] foreach_1.5.2          logging_0.10-108       git2r_0.30.1          
[142] janitor_2.1.0          gridExtra_2.3          farver_2.1.1          
[145] Rtsne_0.16             ggraph_2.0.5           digest_0.6.29         
[148] shiny_1.7.2            networkD3_0.4          Rcpp_1.0.9            
[151] car_3.1-0              broom_1.0.0            performance_0.9.1     
[154] later_1.3.0            sf_1.0-7               ComplexHeatmap_2.10.0 
[157] effectsize_0.7.0       colorspace_2.0-3       brio_1.1.3            
[160] fs_1.5.2               IRanges_2.28.0         splines_4.1.1         
[163] rematch2_2.1.2         graphlayouts_0.8.0     multtest_2.50.0       
[166] sessioninfo_1.2.2      xtable_1.8-4           jsonlite_1.8.0        
[169] nloptr_2.0.3           tidygraph_1.2.1        zeallot_0.1.0         
[172] R6_2.5.1               Hmisc_4.7-0            profvis_0.3.7         
[175] broom.mixed_0.2.9.4    pillar_1.8.0           htmltools_0.5.3       
[178] mime_0.12              glue_1.6.2             fastmap_1.1.0         
[181] minqa_1.2.4            class_7.3-20           codetools_0.2-18      
[184] pkgbuild_1.3.1         mvtnorm_1.1-3          furrr_0.3.0           
[187] utf8_1.2.2             bslib_0.4.0            tibble_3.1.8          
[190] logger_0.2.2           interp_1.1-3           speedyseq_0.5.3.9018  
[193] survival_3.3-1         rmarkdown_2.14         desc_1.4.1            
[196] biomformat_1.22.0      merTools_0.5.2         munsell_0.5.0         
[199] e1071_1.7-11           GetoptLong_1.0.5       rhdf5_2.38.1          
[202] GenomeInfoDbData_1.2.7 iterators_1.0.14       reshape2_1.4.4        
[205] gtable_0.3.0           bayestestR_0.12.1     </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
