<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Marc Noguera-Julian, PhD." />

<meta name="date" content="2022-09-27" />

<title>Microbiome association with Mediterranean diet</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">MedCars</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="201_exploratory_analysis.html">Data Exploration</a>
</li>
<li>
  <a href="210_baseline_analysis.html">Baseline Analysis</a>
</li>
<li>
  <a href="220_intervention_analysis.html">Intervention</a>
</li>
<li>
  <a href="230_MedDietScore.html">Mediterranean Diet</a>
</li>
<li>
  <a href="250_PhysicalActivity.html">Physical Activity</a>
</li>
<li>
  <a href="260_FirmicutesBacteroides.html">Firmicutes/Bacteroidetes</a>
</li>
<li>
  <a href="260_FirmicutesBacteroides.html">Functional Analysis</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Microbiome association with Mediterranean diet</h1>
<h4 class="author">Marc Noguera-Julian, PhD.</h4>
<h4 class="date">2022-09-27</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-09-27
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>2022_MedCars/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version 1.7.0). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20220720code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20220720)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20220720code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20220720)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomMicrobialGenomicsIrsicaixaOrg2022MedCarstree9594b4b1eee834d1666dba133ec4ba2e68d2e3c9targetblank9594b4ba"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/MicrobialGenomics-IrsicaixaOrg/2022_MedCars/tree/9594b4b1eee834d1666dba133ec4ba2e68d2e3c9" target="_blank">9594b4b</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomMicrobialGenomicsIrsicaixaOrg2022MedCarstree9594b4b1eee834d1666dba133ec4ba2e68d2e3c9targetblank9594b4ba" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/MicrobialGenomics-IrsicaixaOrg/2022_MedCars/tree/9594b4b1eee834d1666dba133ec4ba2e68d2e3c9" target="_blank">9594b4b</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    data/.DS_Store
    Ignored:    data/cfu/
    Ignored:    data/internal/
    Ignored:    data/processed/
    Ignored:    data/ps_silva_RR.rds
    Ignored:    data/raw/
    Ignored:    output/.DS_Store
    Ignored:    output/Deliverables/
    Ignored:    output/mre_data/

Untracked files:
    Untracked:  code/902_themes.R

Unstaged changes:
    Modified:   analysis/201_exploratory_analysis.Rmd
    Modified:   analysis/210_baseline_analysis.Rmd
    Modified:   analysis/220_intervention_analysis.Rmd
    Modified:   analysis/300_FunctionalAnalysis.Rmd
    Modified:   code/006_DataPreprocessing.Rmd
    Modified:   code/006_DataPreprocessing.html

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/300_FunctionalAnalysis.Rmd</code>) and HTML (<code>docs/300_FunctionalAnalysis.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/MicrobialGenomics-IrsicaixaOrg/2022_MedCars/blob/1c2dd47bf64409ce32bc85e3c96b5452d29e8fc1/analysis/300_FunctionalAnalysis.Rmd" target="_blank">1c2dd47</a>
</td>
<td>
Marc Noguera-Julian
</td>
<td>
2022-09-14
</td>
<td>
added analysis
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="functional-analysis" class="section level2">
<h2>Functional Analysis</h2>
<p>We are analysing 16S data. This kind of data, by definition, does not contain any sequence other than 16S gene. We can not directly quantify presence of bacterial genes in it. However, we can inferr the abundance of these genes by using bacterial genomics catalogs and bacterial abundance. This is done by Picrust2 and others softwares that take ASV/Genus counts and their representative sequences and looks for probable gene content in genome catalogs.</p>
<div id="picrust2-files-generation" class="section level3">
<h3>PiCrust2 files generation</h3>
<p>First of all we need to transform data located within our R objects and files into somethin that Picrust can use as input.</p>
<pre class="r"><code>suppressWarnings(require(phyloseq))
suppressWarnings(require(biomformat))
ps_silva &lt;- readRDS(&quot;data/cfu/DADA2/ps_silva.rds&quot;)
## Create Biom file
biomformat::write_biom(biomformat::make_biom(data = otu_table(ps_silva) %&gt;%
    as.data.frame(), sample_metadata = phyloseq::sample_data(ps_silva) %&gt;%
    as.data.frame(), tax_table(ps_silva) %&gt;%
    as.data.frame(), id = &quot;MedCars-Microbiome&quot;), biom_file = &quot;data/cfu/DADA2/ps_silva.biom&quot;)

## Create Text file for ASV
my_otu_df &lt;- otu_table(ps_silva, taxa_are_rows = T) %&gt;%
    data.frame()
colnames(my_otu_df) &lt;- colnames(otu_table(ps_silva, taxa_are_rows = T))
readr::write_delim(my_otu_df %&gt;%
    tibble::rownames_to_column(), file = &quot;data/cfu/DADA2/ps_silva_ASV.txt&quot;, delim = &quot;\t&quot;)

## Create Sequence file - need to filter for ASV present in ps_silva
load(&quot;data/raw/DADA2/DADA2_Rsession_raw.RData&quot;)
ps_silva &lt;- readRDS(&quot;data/cfu/DADA2/ps_silva.rds&quot;)
aux_df &lt;- colnames(seqtab.nochim.bkp) %&gt;%
    as.data.frame()
colnames(aux_df) &lt;- &quot;otu_id&quot;
aux_df &lt;- aux_df %&gt;%
    dplyr::filter(otu_id %in% c(ps_silva %&gt;%
        otu_table() %&gt;%
        rownames()))
aux_df$Sequence &lt;- rownames(aux_df)
rownames(aux_df) &lt;- aux_df$otu_id
print(&quot;Creating fasta file&quot;)
file.remove(&quot;data/cfu/DADA2/ps_silva.fas&quot;)
for (i in 1:nrow(aux_df)) {
    write(paste(&quot;&gt;&quot;, aux_df[i, &quot;otu_id&quot;], &quot;\n&quot;, aux_df[i, &quot;Sequence&quot;], sep = &quot;&quot;),
        file = &quot;data/cfu/DADA2/ps_silva.fas&quot;, append = T)
}</code></pre>
</div>
<div id="picrust-run" class="section level3">
<h3>Picrust run</h3>
<p>Now we run picrust and performe the gene content inference. This step takes a while and is only run once to producte the inferred gene content.</p>
<pre class="bash"><code>source activate picrust2 ### Need a conda environement (conda create -n picrust2 -c bioconda picrust2)
picrust2_pipeline.py -s data/cfu/DADA2/ps_silva.fas -i  data/cfu/DADA2/ps_silva_ASV.txt -t epa-ng -o data/cfu/DADA2/picrust2_out_pipeline -p 4</code></pre>
<!--
### Wilcoxon 

Pathways and gene abundance can be tested using Wilcoxon ranked test but we'll opt out of this, because it is heavy computationally and because we have better options-->
</div>
<div id="maaslin---ig" class="section level3">
<h3>MaasLin - IG</h3>
<p>We are interested in detecting the functional change of the gut microbiome after the nutritional intervention. We are using inferred gene content as obtained from picrust2, using 16s data. Interestingly, picrust2 output is obtained either as a gene familly list (KEGG or EC annotated) or as a pathway list using MetaCyc annotation. The latter uses minPath to detect pathway coverage/abundance.</p>
<p>We’ll do both analysis as exploratory. Note that the dimensions of these datasets are large and quite a large number of diufferential abundance genes and pathways are to be expected. We are more interested in either interpreting the iceberg tips of each comparative and to compare the difference sets in IG and CG.</p>
<p>Interestingly, there are much more significantly different pathways and gene familiies vs timepoint within the IG than within the CG.</p>
<div id="pathways" class="section level4">
<h4>Pathways</h4>
<pre><code>suggested package selbal ✔suggested package DataExplorer ✔suggested package lme4 ✔suggested package merTools ✔</code></pre>
<p><img src="figure/300_FunctionalAnalysis.Rmd/DA%20species%20maaslin2%20C01/IG-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="gene-families" class="section level4">
<h4>Gene Families</h4>
<pre><code>suggested package selbal ✔suggested package DataExplorer ✔suggested package lme4 ✔suggested package merTools ✔</code></pre>
<p><img src="figure/300_FunctionalAnalysis.Rmd/DA%20Gene%20Families%20maaslin2%20C01/IG-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="maaslin---cg" class="section level3">
<h3>MaasLin - CG</h3>
<div id="pathways-1" class="section level4">
<h4>Pathways</h4>
<pre><code>suggested package selbal ✔suggested package DataExplorer ✔suggested package lme4 ✔suggested package merTools ✔</code></pre>
<p><img src="figure/300_FunctionalAnalysis.Rmd/DA%20species%20maaslin2%20C02/CG-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="gene-families-1" class="section level4">
<h4>Gene Families</h4>
<pre><code>suggested package selbal ✔suggested package DataExplorer ✔suggested package lme4 ✔suggested package merTools ✔</code></pre>
<p><img src="figure/300_FunctionalAnalysis.Rmd/DA%20Gene%20Families%20maaslin2%20C02/CG-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="enrichment-analysis" class="section level3">
<h3>Enrichment Analysis</h3>
<p>In order to further interpret the list of gene families that are either over or under-represented in Month6 respect to Baseline in each of the CG and IG we need to perform some enrichment based analysis that tells us which molecular modules or pathways are over-represented <img src="figure/300_FunctionalAnalysis.Rmd/enrichment%20analysis%20IG-1.png" width="672" style="display: block; margin: auto;" /><img src="figure/300_FunctionalAnalysis.Rmd/enrichment%20analysis%20IG-2.png" width="672" style="display: block; margin: auto;" /></p>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre><code>R version 4.1.1 (2021-08-10)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Big Sur 10.16

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.1/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
 [1] clusterProfiler_4.2.2 forcats_0.5.1         stringr_1.4.1        
 [4] purrr_0.3.4           readr_2.1.2           tidyr_1.2.0          
 [7] tibble_3.1.8          ggplot2_3.3.6         tidyverse_1.3.2      
[10] metar_0.1.5           scales_1.2.1          testthat_3.1.4       
[13] dplyr_1.0.9           ggstatsplot_0.9.3     vegan_2.6-2          
[16] lattice_0.20-45       permute_0.9-7         patchwork_1.1.1      
[19] phylosmith_1.0.6      phyloseq_1.38.0      

loaded via a namespace (and not attached):
  [1] class_7.3-20           ps_1.7.1               foreach_1.5.2         
  [4] rprojroot_2.0.3        crayon_1.5.1           MASS_7.3-58           
  [7] rhdf5filters_1.6.0     nlme_3.1-158           backports_1.4.1       
 [10] reprex_2.0.1           GOSemSim_2.20.0        rlang_1.0.4           
 [13] XVector_0.34.0         readxl_1.4.0           performance_0.9.1     
 [16] nloptr_2.0.3           callr_3.7.1            BiocParallel_1.28.3   
 [19] rjson_0.2.21           bit64_4.0.5            glue_1.6.2            
 [22] parallel_4.1.1         processx_3.7.0         AnnotationDbi_1.56.2  
 [25] BiocGenerics_0.40.0    classInt_0.4-7         DOSE_3.20.1           
 [28] haven_2.5.0            tidyselect_1.1.2       usethis_2.1.6         
 [31] sf_1.0-7               ggpubr_0.4.0           xtable_1.8-4          
 [34] magrittr_2.0.3         evaluate_0.15          cli_3.3.0             
 [37] zlibbioc_1.40.0        DataExplorer_0.8.2     rstudioapi_0.13       
 [40] miniUI_0.1.1.1         furrr_0.3.0            whisker_0.4           
 [43] logger_0.2.2           bslib_0.4.0            fastmatch_1.1-3       
 [46] treeio_1.18.1          shiny_1.7.2            xfun_0.31             
 [49] clue_0.3-61            parameters_0.18.1      pkgbuild_1.3.1        
 [52] multtest_2.50.0        cluster_2.1.3          tidygraph_1.2.1       
 [55] biomformat_1.22.0      KEGGREST_1.34.0        ggrepel_0.9.1         
 [58] logging_0.10-108       ape_5.6-2              listenv_0.8.0         
 [61] Biostrings_2.62.0      png_0.1-7              future_1.27.0         
 [64] zeallot_0.1.0          withr_2.5.0            bitops_1.0-7          
 [67] ggforce_0.3.3          plyr_1.8.7             cellranger_1.1.0      
 [70] e1071_1.7-11           coda_0.19-4            pillar_1.8.1          
 [73] RcppParallel_5.1.5     GlobalOptions_0.1.2    cachem_1.0.6          
 [76] fs_1.5.2               GetoptLong_1.0.5       paletteer_1.4.0       
 [79] vctrs_0.4.1            ellipsis_0.3.2         generics_0.1.3        
 [82] devtools_2.4.4         tools_4.1.1            munsell_0.5.0         
 [85] tweenr_1.0.2           fgsea_1.20.0           networkD3_0.4         
 [88] proxy_0.4-27           fastmap_1.1.0          compiler_4.1.1        
 [91] pkgload_1.3.0          abind_1.4-5            httpuv_1.6.5          
 [94] sessioninfo_1.2.2      GenomeInfoDbData_1.2.7 gridExtra_2.3         
 [97] workflowr_1.7.0        utf8_1.2.2             later_1.3.0           
[100] selbal_0.1.0           jsonlite_1.8.0         arm_1.12-2            
[103] tidytree_0.3.9         carData_3.0-5          lazyeval_0.2.2        
[106] promises_1.2.0.1       car_3.1-0              doParallel_1.0.17     
[109] rmarkdown_2.14         textshaping_0.3.6      blme_1.0-5            
[112] Rtsne_0.16             downloader_0.4         Biobase_2.54.0        
[115] igraph_1.3.4           survival_3.3-1         yaml_2.3.5            
[118] systemfonts_1.0.4      htmltools_0.5.3        memoise_2.0.1         
[121] profvis_0.3.7          graphlayouts_0.8.0     IRanges_2.28.0        
[124] here_1.0.1             viridisLite_0.4.1      digest_0.6.29         
[127] assertthat_0.2.1       mime_0.12              wesanderson_0.3.6     
[130] bayestestR_0.12.1      units_0.8-0            RSQLite_2.2.15        
[133] yulab.utils_0.0.5      remotes_2.4.2          data.table_1.14.2     
[136] urlchecker_1.0.1       blob_1.2.3             S4Vectors_0.32.4      
[139] ragg_1.2.2             splines_4.1.1          labeling_0.4.2        
[142] rematch2_2.1.2         Rhdf5lib_1.16.0        googledrive_2.0.0     
[145] RCurl_1.98-1.8         broom_1.0.0            hms_1.1.2             
[148] modelr_0.1.8           rhdf5_2.38.1           colorspace_2.0-3      
[151] shape_1.4.6            aplot_0.1.6            broom.mixed_0.2.9.4   
[154] sass_0.4.2             Rcpp_1.0.9             mvtnorm_1.1-3         
[157] circlize_0.4.15        enrichplot_1.14.2      fansi_1.0.3           
[160] tzdb_0.3.0             brio_1.1.3             parallelly_1.32.1     
[163] R6_2.5.1               grid_4.1.1             lifecycle_1.0.1       
[166] formatR_1.12           statsExpressions_1.3.2 datawizard_0.4.1      
[169] ggsignif_0.6.3         googlesheets4_1.0.0    minqa_1.2.4           
[172] jquerylib_0.1.4        DO.db_2.9              Matrix_1.4-1          
[175] qvalue_2.26.0          desc_1.4.1             RColorBrewer_1.1-3    
[178] iterators_1.0.14       htmlwidgets_1.5.4      polyclip_1.10-0       
[181] shadowtext_0.1.2       gridGraphics_0.5-1     rvest_1.0.2           
[184] ComplexHeatmap_2.10.0  mgcv_1.8-40            globals_0.15.1        
[187] insight_0.18.0         codetools_0.2-18       matrixStats_0.62.0    
[190] lubridate_1.8.0        GO.db_3.14.0           prettyunits_1.1.1     
[193] dbplyr_2.2.1           GenomeInfoDb_1.30.1    correlation_0.8.1     
[196] gtable_0.3.0           DBI_1.1.3              git2r_0.30.1          
[199] stats4_4.1.1           ggfun_0.0.6            httr_1.4.3            
[202] highr_0.9              KernSmooth_2.23-20     stringi_1.7.8         
[205] vroom_1.5.7            reshape2_1.4.4         farver_2.1.1          
[208] viridis_0.6.2          ggtree_3.2.1           xml2_1.3.3            
[211] boot_1.3-28            lme4_1.1-30            ade4_1.7-19           
[214] ggplotify_0.1.0        bit_4.0.4              scatterpie_0.1.7      
[217] ggside_0.2.1           ggraph_2.0.5           pkgconfig_2.0.3       
[220] gargle_1.2.0           rstatix_0.7.0          knitr_1.39            
[223] merTools_0.5.2        </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
