---
site: workflowr::wflow_site
title: "Nutritional workcenter intervention based on mediterranean diet linked to microbiome changes in the gut microbiome"
author: "Marc Noguera-Julian, PhD. "
date: "`r format(Sys.time(), '%Y-%m-%d')`"
header-includes:
- \usepackage{fancyhdr}
- \usepackage{lipsum}
- \pagestyle{fancy}
- \fancyhead[CO,CE]{Microbioma intestinal en intervencion nutricional}
- \fancyfoot[LO,RE]{Marc Noguera-Julian, IrsiCaixa}
- \fancyhead[LE,RO]{\thepage}
- \fancyfoot[CO,CE]{}
- \usepackage[default]{sourcesanspro}
- \usepackage[T1]{fontenc}
- \usepackage{eso-pic,graphicx,transparent}
mainfont: Helvetica
output: html_document
---

## Introduction   

This is the shared analysis workflow for the MedCars study, microbiome substudy. MedCars is a nutritional intervention study based on the mediterranean diet. Participants are health individuals between 18 and 65 years of age that work at either working center (C01 or C02). The clustered randomized design results in focusing the nutritional intervention in one of two workcenters, while the other serves as control.  C01 Serves as Intervention Group (IG) hile C02 serves as Control Group(CG)

Nutritional intervention is based on different actions taken at the workplace, some of them being common to both IG and CG or exclusive to IG:

 * High fiber bread sandwich, instead of standard sandwich (IG).
 * 5L Extra Virgin Olive Oil(EVOO) given to each participant each month (IG).
 * 30 g/day nuts(IG).
 * Healthy food in vending machines(IG).
 * Nutritional counseling for a healthy lifestyle(CG & IG)
 * Psychological counselling(CG & IG).
 * 6 month follow-up(CG & IG).
 
Fecal samples are taken before intervention start and after 6 month follow-up. Gut microbiome is characterized using standard procedures for 16s rDNA sequencing.

## Methods

### Microbiome Analysis

Genomic DNA from all samples was extracted utilizing the Qiagen Powerfecal DNA Kit (Cat # 12888–50) and quantified via Quant-iT PicoGreen dsDNA Assay (Cat # P11496). The 16s V3-V4 region was amplified through 30 cycles PCR with an F515 locus specific primer (5’-TCGTCGGCAGCGTCAGA-TGTGTATAAGAGACAG-3’) and an 806R-locus specific primer (5’-GTCTCGTGGGC-TCGGAGATGTGTATAAGAGACAG-3’) both with compatible overhang regions for index attachment. [22] PCR cleanup was performed using Agencourt AmpureXP magnetic beads (Cat # A63880). V3-V4 fragments for all samples were indexed utilizing a second PCR reaction with Illumina Nextera XT Index Kit (Cat # FC-121–1012) and index-adhering primers (1.1: 5’- AATGATACGGCGACCACCGAGAT-3’; 2.1: 5’-CAAGCAGAAGACGGCATACGA-3’). Indexed fragments were purified with Ampure XP beads, quantified with dsDNA Picogreen, and pooled with respect to achieving equal molecular concentrations for all samples in the final library. About 300 paired-end sequencing was performed on the MiSeq platform using an Illumina MiSeq Version 2 500-cycle Kit (Cat # MS-102–2003) at a loading concentration of 10 picomolar. An Illumina PhiX control (Cat # FC-110–3001) was utilized in all MiSeq runs as recommended in the MiSeq System User Guide for metagenomic samples.

### 16S Data Analysis

Raw fastq files were process within the R::DADA2 pipeline. We used default parameters except in the filtering step when we set maxEE to 2, 10 to account for the lower quality of reverse reads inherent in Illumina technology. We removed chimeric reads using the removeBimeraDenovo function in the same package. ASV sequences were annotated using RDP algorithm against Silva database. ASV representative sequences were used to build and alignment with MAFFT and the corresponding phylogenetic tree with FastTree. Downstream analysis was based on phyloseq v1.22.39. All ASVs assigned to neither Bacteria nor Archaea or not present in at least 1% of the samples were removed.

### Statistical Testing: Alpha/Beta Diversity and Differential Abundance.

We calculated Chao1 and Shannon diversity index for every sample using the phyloseq estimate_richness function. Phylogenetic distance was calculated using R::Picante package. We performed t-test/Wilcoxon to test for differences in richness between groups of interest. Weighted unifrac distances based on the calculated phylogenetic tree were used as microbiome beta-diversity measures for the calculation on non-metric multidimensional analysis (NMDS) and permanova tests. R::LEFsEr packages was used to perform all tests on differential abundance between groups. 

